<!DOCTYPE html><html lang="en"><head><meta charSet="utf-8"/><meta http-equiv="x-ua-compatible" content="ie=edge"/><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"/><meta name="generator" content="Gatsby 5.6.0"/><meta data-react-helmet="true" itemProp="name" content="Simulating sequencing datasets"/><meta data-react-helmet="true" itemProp="description" content="Bioinformatics tools and pipelines, as with any analysis methods, need to be tested to ensure they…"/><meta data-react-helmet="true" name="description" content="Bioinformatics tools and pipelines, as with any analysis methods, need to be tested to ensure they…"/><meta data-react-helmet="true" property="og:title" content="Simulating sequencing datasets"/><meta data-react-helmet="true" property="og:description" content="Bioinformatics tools and pipelines, as with any analysis methods, need to be tested to ensure they…"/><meta data-react-helmet="true" property="og:type" content="article"/><meta data-react-helmet="true" property="og:site_name" content="EPI2ME"/><meta data-react-helmet="true" property="og:image" content="https://epi2me.nanoporetech.com/static/676ccc6f1894f739f61cc237322c970e/23364/polar_bear.jpg"/><meta data-react-helmet="true" name="twitter:card" content="summary"/><meta data-react-helmet="true" name="twitter:site" content="EPI2ME"/><meta data-react-helmet="true" name="twitter:title" content="Simulating sequencing datasets"/><meta data-react-helmet="true" name="twitter:description" content="Bioinformatics tools and pipelines, as with any analysis methods, need to be tested to ensure they…"/><meta data-react-helmet="true" name="twitter:creator" content="https://twitter.com/nanopore"/><meta data-react-helmet="true" name="keywords" content="epi2melabs, simulation, benchmarking"/><meta data-react-helmet="true" name="article:published_time" content="March 18, 2024"/><meta data-react-helmet="true" name="twitter:label1" value="Reading time"/><meta data-react-helmet="true" name="twitter:data1" value="12 min read"/><meta name="theme-color" content="#0084A9"/><style data-href="/styles.b5ef6fa9d6ffaed9b56b.css" data-identity="gatsby-global-css">.DrawerPortalOverlay.ReactModal__Overlay{background:rgba(0,0,0,.75);height:100vh;left:0;opacity:0;outline:none;position:fixed;top:0;width:100%;z-index:20}.DrawerPortalOverlay.ReactModal__Overlay--after-open{opacity:1;transition:all .3s cubic-bezier(.77,0,.175,1)}.DrawerPortalOverlay.ReactModal__Overlay--before-close{opacity:0;transition:all .15s cubic-bezier(.77,0,.175,1);transition-delay:.15s}.DrawerPortalContent.ReactModal__Content{height:100vh;margin-left:auto;max-width:650px;outline:none;transform:translate(100%);transition:all .3s cubic-bezier(.78,.14,.15,.86);width:80%}.DrawerPortalContent.ReactModal__Content--after-open{box-shadow:0 0 45px 0 rgba(0,0,0,.25);transform:translate(0)}.DrawerPortalContent.ReactModal__Content--before-close{transform:translate(100%);transition:all .3s cubic-bezier(.78,.14,.15,.86)}.ModalPortalOverlay{background:rgba(0,0,0,.75);height:100%;left:0;opacity:0;outline:none;position:fixed;top:0;width:100%;z-index:20}.ModalPortalOverlay.ReactModal__Overlay--after-open{opacity:1;transition:all .3s cubic-bezier(.77,0,.175,1)}.ModalPortalOverlay.ReactModal__Overlay--before-close{opacity:0;transition:all .3s cubic-bezier(.77,0,.175,1)}.ModalPortalContent{left:50%;outline:none;position:relative;top:50%;transform:scale(.7) translate(-50%,-50%);transform-origin:left top;transition:transform .3s cubic-bezier(.78,.14,.15,.86);width:100%}.ModalPortalContent.ReactModal__Content--after-open{box-shadow:0 0 45px 0 rgba(0,0,0,.25);transform:scale(1) translate(-50%,-50%)}.ModalPortalContent.ReactModal__Content--before-close{transform:scale(1.1) translate(-50%,-50%)}.ReactModal__Body--open,.ReactModal__Html--open,.rc-switch{overflow:hidden}.rc-switch{background-color:#ccc;border:1px solid #ccc;border-radius:20px 20px;box-sizing:border-box;cursor:pointer;display:inline-block;height:22px;line-height:20px;padding:0;position:relative;transition:all .3s cubic-bezier(.35,0,.25,1);vertical-align:middle;width:44px}.rc-switch-inner-checked,.rc-switch-inner-unchecked{color:#fff;font-size:12px;position:absolute;top:0;transition:left .3s cubic-bezier(.35,0,.25,1)}.rc-switch-inner-checked{left:-14px}.rc-switch-inner-unchecked{left:24px}.rc-switch:after{animation-duration:.3s;animation-name:rcSwitchOff;animation-timing-function:cubic-bezier(.35,0,.25,1);background-color:#fff;border-radius:50% 50%;box-shadow:0 2px 5px rgba(0,0,0,.26);content:" ";cursor:pointer;height:18px;left:2px;position:absolute;top:1px;transform:scale(1);transition:left .3s cubic-bezier(.35,0,.25,1);width:18px}.rc-switch:hover:after{animation-name:rcSwitchOn;transform:scale(1.1)}.rc-switch:focus{box-shadow:0 0 0 2px #d5f1fd;outline:none}.rc-switch-checked{background-color:#87d068;border:1px solid #87d068}.rc-switch-checked .rc-switch-inner-checked{left:6px}.rc-switch-checked .rc-switch-inner-unchecked{left:44px}.rc-switch-checked:after{left:22px}.rc-switch-disabled{background:#ccc;border-color:#ccc;cursor:no-drop}.rc-switch-disabled:after{animation-name:none;background:#9e9e9e;cursor:no-drop}.rc-switch-disabled:hover:after{animation-name:none;transform:scale(1)}.rc-switch-label{display:inline-block;font-size:14px;line-height:20px;padding-left:10px;pointer-events:none;-webkit-user-select:text;user-select:text;vertical-align:middle;white-space:normal}@keyframes rcSwitchOn{0%{transform:scale(1)}50%{transform:scale(1.25)}to{transform:scale(1.1)}}@keyframes rcSwitchOff{0%{transform:scale(1.1)}to{transform:scale(1)}}@keyframes fadeInUp{0%{opacity:0;transform:translate3d(0,10%,0)}to{opacity:1;transform:translateZ(0)}}@keyframes fadeInDown{0%{opacity:0;transform:translate3d(0,-10%,0)}to{opacity:1;transform:translateZ(0)}}@keyframes fadeInLeft{0%{opacity:0;transform:translate3d(-20%,0,0)}to{opacity:1;transform:translateZ(0)}}@keyframes fadeInRight{0%{opacity:0;transform:translate3d(20%,0,0)}to{opacity:1;transform:translateZ(0)}}@keyframes fadeIn{0%{opacity:0}to{opacity:1}}@media screen and (min-width:768px){.react-reveal{animation-timing-function:cubic-bezier(.77,0,.175,1)}.fadeInUp{animation-name:fadeInUp}.fadeInDown{animation-name:fadeInDown}.fadeInLeft{animation-name:fadeInLeft}.fadeInRight{animation-name:fadeInRight}.fadeIn{animation-name:fadeIn}}@media not screen and (min-width:768px){.react-reveal{opacity:1!important}}.slick-slider{-webkit-touch-callout:none;-webkit-tap-highlight-color:transparent;box-sizing:border-box;touch-action:pan-y;-webkit-user-select:none;user-select:none;-khtml-user-select:none}.slick-list,.slick-slider{display:block;position:relative}.slick-list{margin:0;overflow:hidden;padding:0}.slick-list:focus{outline:none}.slick-list.dragging{cursor:pointer;cursor:hand}.slick-slider .slick-list,.slick-slider .slick-track{transform:translateZ(0)}.slick-track{display:block;left:0;margin-left:auto;margin-right:auto;position:relative;top:0}.slick-track:after,.slick-track:before{content:"";display:table}.slick-track:after{clear:both}.slick-loading .slick-track{visibility:hidden}.slick-slide{display:none;float:left;height:100%;min-height:1px}[dir=rtl] .slick-slide{float:right}.slick-slide img{display:block}.slick-slide.slick-loading img{display:none}.slick-slide.dragging img{pointer-events:none}.slick-initialized .slick-slide{display:block}.slick-loading .slick-slide{visibility:hidden}.slick-vertical .slick-slide{border:1px solid transparent;display:block;height:auto}.slick-arrow.slick-hidden{display:none}.with-animation .slick-slide{opacity:1;transform:scale(1);transform-origin:left;transition:all .6s ease}.with-animation .slick-slide.slick-active+:not(.slick-active){opacity:0;transform:scale(.7)}.slick-next,.slick-prev{position:absolute;top:50%;transform:translateY(-50%);z-index:2}.slick-prev{box-shadow:-1px 1px 3px 0 rgba(1,1,1,.1);left:-15px}.slick-next{box-shadow:1px 1px 3px 0 rgba(1,1,1,.1);right:-15px}.slick-dots{display:block;list-style:none;margin:0;padding:.5rem 0 0;position:absolute;text-align:center;text-align:right;top:-4rem;width:100%}.slick-dots li{cursor:pointer;display:inline-block;margin:0 5px;padding:0;position:relative}.slick-dots li button{background:transparent;background-color:#a0aec0;border:0;border-radius:100%;cursor:pointer;display:block;font-size:0;height:12px;line-height:0px;margin:5px;outline:none;width:12px}.slick-dots li.slick-active button{background-color:#4a5568}</style><link rel="sitemap" type="application/xml" href="/sitemap-index.xml"/><style>.gatsby-image-wrapper{position:relative;overflow:hidden}.gatsby-image-wrapper picture.object-fit-polyfill{position:static!important}.gatsby-image-wrapper img{bottom:0;height:100%;left:0;margin:0;max-width:none;padding:0;position:absolute;right:0;top:0;width:100%;object-fit:cover}.gatsby-image-wrapper [data-main-image]{opacity:0;transform:translateZ(0);transition:opacity .25s linear;will-change:opacity}.gatsby-image-wrapper-constrained{display:inline-block;vertical-align:top}</style><noscript><style>.gatsby-image-wrapper noscript [data-main-image]{opacity:1!important}.gatsby-image-wrapper [data-placeholder-image]{opacity:0!important}</style></noscript><script type="module">const e="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;e&&document.body.addEventListener("load",(function(e){const t=e.target;if(void 0===t.dataset.mainImage)return;if(void 0===t.dataset.gatsbyImageSsr)return;let a=null,n=t;for(;null===a&&n;)void 0!==n.parentNode.dataset.gatsbyImageWrapper&&(a=n.parentNode),n=n.parentNode;const o=a.querySelector("[data-placeholder-image]"),r=new Image;r.src=t.currentSrc,r.decode().catch((()=>{})).then((()=>{t.style.opacity=1,o&&(o.style.opacity=0,o.style.transition="opacity 500ms linear")}))}),!0);</script><link crossorigin="" href="https://fonts.gstatic.com/" rel="preconnect"/><link crossorigin="" href="https://fonts.googleapis.com/" rel="preconnect"/><title data-react-helmet="true">Simulating sequencing datasets | EPI2ME Blog</title><script data-react-helmet="true" type="text/javascript">
            window.heap=window.heap||[],heap.load=function(e,t){window.heap.appid=e,window.heap.config=t=t||{};var r=document.createElement("script");r.type="text/javascript",r.async=!0,r.src="https://cdn.heapanalytics.com/js/heap-"+e+".js";var a=document.getElementsByTagName("script")[0];a.parentNode.insertBefore(r,a);for(var n=function(e){return function(){heap.push([e].concat(Array.prototype.slice.call(arguments,0)))}},p=["addEventProperties","addUserProperties","clearEventProperties","identify","resetIdentity","removeEventProperty","setEventProperties","track","unsetEventProperty"],o=0;o<p.length;o++)heap[p[o]]=n(p[o])};
        </script><script data-react-helmet="true" type="application/ld+json">{"@context":"https://schema.org","@type":"Article","headline":"Simulating sequencing datasets","image":"https://epi2me.nanoporetech.com/static/676ccc6f1894f739f61cc237322c970e/23364/polar_bear.jpg","datePublished":"March 18, 2024"}</script><script data-react-helmet="true" type="application/ld+json">{"@context":"https://schema.org","@type":"BreadcrumbList","itemListElement":[{"@type":"ListItem","position":1,"name":"EPI2ME","item":"https://epi2me.nanoporetech.com"},{"@type":"ListItem","position":2,"name":"How Tos","item":"https://epi2me.nanoporetech.com/category/how-tos/"}]}</script><link rel="icon" href="/favicon-32x32.png?v=4f61fcbcea1f2514aba501fcda222617" type="image/png"/><link rel="manifest" href="/manifest.webmanifest" crossorigin="anonymous"/><link rel="apple-touch-icon" sizes="48x48" href="/icons/icon-48x48.png?v=4f61fcbcea1f2514aba501fcda222617"/><link rel="apple-touch-icon" sizes="72x72" href="/icons/icon-72x72.png?v=4f61fcbcea1f2514aba501fcda222617"/><link rel="apple-touch-icon" sizes="96x96" href="/icons/icon-96x96.png?v=4f61fcbcea1f2514aba501fcda222617"/><link rel="apple-touch-icon" sizes="144x144" href="/icons/icon-144x144.png?v=4f61fcbcea1f2514aba501fcda222617"/><link rel="apple-touch-icon" sizes="192x192" href="/icons/icon-192x192.png?v=4f61fcbcea1f2514aba501fcda222617"/><link rel="apple-touch-icon" sizes="256x256" href="/icons/icon-256x256.png?v=4f61fcbcea1f2514aba501fcda222617"/><link rel="apple-touch-icon" sizes="384x384" href="/icons/icon-384x384.png?v=4f61fcbcea1f2514aba501fcda222617"/><link rel="apple-touch-icon" sizes="512x512" href="/icons/icon-512x512.png?v=4f61fcbcea1f2514aba501fcda222617"/><link rel="alternate" type="application/rss+xml" title="EPI2ME Labs&#x27;s RSS Feed" href="/rss.xml"/></head><body><div id="___gatsby"><div style="outline:none" tabindex="-1" id="gatsby-focus-wrapper"><style data-emotion-css="g7so3b">body{--theme-ui-colors-text:var(--theme-ui-colors-text,#cbd5e0);--theme-ui-colors-article:var(--theme-ui-colors-article,#e2e8f0);--theme-ui-colors-heading:var(--theme-ui-colors-heading,#edf2f7);--theme-ui-colors-alphaLighter:var(--theme-ui-colors-alphaLighter,#434190);--theme-ui-colors-alphaLight:var(--theme-ui-colors-alphaLight,#90C5E7);--theme-ui-colors-alpha:var(--theme-ui-colors-alpha,#0084A9);--theme-ui-colors-alphaDark:var(--theme-ui-colors-alphaDark,#a3bffa);--theme-ui-colors-alphaDarker:var(--theme-ui-colors-alphaDarker,#ebf4ff);--theme-ui-colors-betaLighter:var(--theme-ui-colors-betaLighter,#553c9a);--theme-ui-colors-betaLight:var(--theme-ui-colors-betaLight,#805ad5);--theme-ui-colors-beta:var(--theme-ui-colors-beta,#0084A9);--theme-ui-colors-betaDark:var(--theme-ui-colors-betaDark,#d6bcfa);--theme-ui-colors-betaDarker:var(--theme-ui-colors-betaDarker,#faf5ff);--theme-ui-colors-omegaLighter:var(--theme-ui-colors-omegaLighter,#1a202c);--theme-ui-colors-omegaLight:var(--theme-ui-colors-omegaLight,#001A21);--theme-ui-colors-omega:var(--theme-ui-colors-omega,#a0aec0);--theme-ui-colors-omegaDark:var(--theme-ui-colors-omegaDark,#e2e8f0);--theme-ui-colors-omegaDarker:var(--theme-ui-colors-omegaDarker,#a0aec0);--theme-ui-colors-successLight:var(--theme-ui-colors-successLight,#f0fff4);--theme-ui-colors-success:var(--theme-ui-colors-success,#48bb78);--theme-ui-colors-errorLight:var(--theme-ui-colors-errorLight,#fff5f5);--theme-ui-colors-error:var(--theme-ui-colors-error,#f56565);--theme-ui-colors-white:var(--theme-ui-colors-white,#fff);--theme-ui-colors-background:var(--theme-ui-colors-background,#1a202c);--theme-ui-colors-contentBg:var(--theme-ui-colors-contentBg,#2d3748);--theme-ui-colors-headerBg:var(--theme-ui-colors-headerBg,#1a202c);--theme-ui-colors-footerBg:var(--theme-ui-colors-footerBg,#2d3748);--theme-ui-colors-mute:var(--theme-ui-colors-mute,#718096);--theme-ui-colors-highlight:var(--theme-ui-colors-highlight,#edf2f7);color:var(--theme-ui-colors-text,#cbd5e0);background-color:var(--theme-ui-colors-background,#1a202c);}body.theme-ui-dark{--theme-ui-colors-alphaLighter:var(--theme-ui-colors-modes-dark-alphaLighter,#434190);--theme-ui-colors-alphaLight:var(--theme-ui-colors-modes-dark-alphaLight,#90C5E7);--theme-ui-colors-alphaDark:var(--theme-ui-colors-modes-dark-alphaDark,#a3bffa);--theme-ui-colors-alphaDarker:var(--theme-ui-colors-modes-dark-alphaDarker,#ebf4ff);--theme-ui-colors-betaLighter:var(--theme-ui-colors-modes-dark-betaLighter,#553c9a);--theme-ui-colors-betaLight:var(--theme-ui-colors-modes-dark-betaLight,#805ad5);--theme-ui-colors-betaDark:var(--theme-ui-colors-modes-dark-betaDark,#d6bcfa);--theme-ui-colors-betaDarker:var(--theme-ui-colors-modes-dark-betaDarker,#faf5ff);--theme-ui-colors-omegaLighter:var(--theme-ui-colors-modes-dark-omegaLighter,#1a202c);--theme-ui-colors-omegaLight:var(--theme-ui-colors-modes-dark-omegaLight,#001A21);--theme-ui-colors-omegaDark:var(--theme-ui-colors-modes-dark-omegaDark,#e2e8f0);--theme-ui-colors-omegaDarker:var(--theme-ui-colors-modes-dark-omegaDarker,#a0aec0);--theme-ui-colors-text:var(--theme-ui-colors-modes-dark-text,#cbd5e0);--theme-ui-colors-article:var(--theme-ui-colors-modes-dark-article,#e2e8f0);--theme-ui-colors-heading:var(--theme-ui-colors-modes-dark-heading,#edf2f7);--theme-ui-colors-background:var(--theme-ui-colors-modes-dark-background,#1a202c);--theme-ui-colors-contentBg:var(--theme-ui-colors-modes-dark-contentBg,#2d3748);--theme-ui-colors-headerBg:var(--theme-ui-colors-modes-dark-headerBg,#1a202c);--theme-ui-colors-footerBg:var(--theme-ui-colors-modes-dark-footerBg,#2d3748);--theme-ui-colors-mute:var(--theme-ui-colors-modes-dark-mute,#718096);}</style><style data-emotion-css="wxng5j">*{box-sizing:border-box;}body{margin:0;font-family:'Inter',sans-serif;font-size:1rem;font-weight:normal;line-height:1.8;-webkit-font-smoothing:antialiased;-webkit-text-decoration:none;text-decoration:none;}</style><style data-emotion-css="130izn8">.css-130izn8{box-sizing:border-box;margin:0;min-width:0;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;height:100vh;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}</style><div class="css-130izn8"><style data-emotion-css="18ygtip">html{font-size:85%;-webkit-scroll-behavior:smooth;-moz-scroll-behavior:smooth;-ms-scroll-behavior:smooth;scroll-behavior:smooth;}@media screen and (min-width:640px){html{font-size:90%;}}@media screen and (min-width:900px){html{font-size:95%;}}@media screen and (min-width:1400px){html{font-size:100%;}}::selection{color:var(--theme-ui-colors-white,#fff);background:var(--theme-ui-colors-omegaDark,#e2e8f0);}</style><style data-emotion-css="vqott">.css-vqott{box-sizing:border-box;margin:0;min-width:0;position:relative;background-color:var(--theme-ui-colors-headerBg,#1a202c);}</style><div class="css-vqott"><style data-emotion-css="gfo1yo">.css-gfo1yo{box-sizing:border-box;margin:0;min-width:0;width:100%;max-width:1400px;margin-left:auto;margin-right:auto;padding-left:1rem;padding-right:1rem;position:relative;z-index:10;}@media screen and (min-width:640px){.css-gfo1yo{padding-left:2rem;padding-right:2rem;}}</style><div class="css-gfo1yo"><style data-emotion-css="16oqm0b">.css-16oqm0b{box-sizing:border-box;margin:0;min-width:0;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;height:6rem;padding-top:1rem;padding-bottom:1rem;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}@media screen and (min-width:640px){.css-16oqm0b{height:7rem;padding-top:2rem;padding-bottom:2rem;}}</style><div class="css-16oqm0b"><style data-emotion-css="1cw1z63">.css-1cw1z63{box-sizing:border-box;margin:0;min-width:0;-webkit-flex-basis:100%;-ms-flex-preferred-size:100%;flex-basis:100%;min-width:auto;}@media screen and (min-width:900px){.css-1cw1z63{-webkit-flex-basis:16.666667%;-ms-flex-preferred-size:16.666667%;flex-basis:16.666667%;}}</style><div class="css-1cw1z63"><style data-emotion-css="1b1vmhm">.css-1b1vmhm{box-sizing:border-box;margin:0;min-width:0;font-family:'Inter',sans-serif;font-weight:bold;line-height:1.33;display:block;color:var(--theme-ui-colors-heading,#edf2f7);font-weight:bold;-webkit-text-decoration:none;text-decoration:none;margin-bottom:1rem;font-size:1.5rem;margin:0;}</style><a alt="EPI2ME Blog" aria-label="EPI2ME Blog" class="css-1b1vmhm" href="/"><div data-gatsby-image-wrapper="" style="vertical-align:middle" class="gatsby-image-wrapper gatsby-image-wrapper-constrained"><div style="max-width:150px;display:block"><img alt="" role="presentation" aria-hidden="true" src="data:image/svg+xml;charset=utf-8,%3Csvg%20height=&#x27;41&#x27;%20width=&#x27;150&#x27;%20xmlns=&#x27;http://www.w3.org/2000/svg&#x27;%20version=&#x27;1.1&#x27;%3E%3C/svg%3E" style="max-width:100%;display:block;position:static"/></div><div aria-hidden="true" data-placeholder-image="" style="opacity:1;transition:opacity 500ms linear"></div><picture><source type="image/webp" srcSet="/static/dbb47454a74ffdac6ab079de768cc8af/19305/logo-dark.webp 38w,/static/dbb47454a74ffdac6ab079de768cc8af/95b49/logo-dark.webp 75w,/static/dbb47454a74ffdac6ab079de768cc8af/742ee/logo-dark.webp 150w,/static/dbb47454a74ffdac6ab079de768cc8af/3421a/logo-dark.webp 300w" sizes="(min-width: 150px) 150px, 100vw"/><img data-gatsby-image-ssr="" data-main-image="" style="opacity:0" sizes="(min-width: 150px) 150px, 100vw" decoding="async" loading="eager" src="/static/dbb47454a74ffdac6ab079de768cc8af/8160b/logo-dark.png" srcSet="/static/dbb47454a74ffdac6ab079de768cc8af/93842/logo-dark.png 38w,/static/dbb47454a74ffdac6ab079de768cc8af/9c49a/logo-dark.png 75w,/static/dbb47454a74ffdac6ab079de768cc8af/8160b/logo-dark.png 150w,/static/dbb47454a74ffdac6ab079de768cc8af/4994d/logo-dark.png 300w" alt="EPI2ME Blog"/></picture><noscript><picture><source type="image/webp" srcSet="/static/dbb47454a74ffdac6ab079de768cc8af/19305/logo-dark.webp 38w,/static/dbb47454a74ffdac6ab079de768cc8af/95b49/logo-dark.webp 75w,/static/dbb47454a74ffdac6ab079de768cc8af/742ee/logo-dark.webp 150w,/static/dbb47454a74ffdac6ab079de768cc8af/3421a/logo-dark.webp 300w" sizes="(min-width: 150px) 150px, 100vw"/><img data-gatsby-image-ssr="" data-main-image="" style="opacity:0" sizes="(min-width: 150px) 150px, 100vw" decoding="async" loading="eager" src="/static/dbb47454a74ffdac6ab079de768cc8af/8160b/logo-dark.png" srcSet="/static/dbb47454a74ffdac6ab079de768cc8af/93842/logo-dark.png 38w,/static/dbb47454a74ffdac6ab079de768cc8af/9c49a/logo-dark.png 75w,/static/dbb47454a74ffdac6ab079de768cc8af/8160b/logo-dark.png 150w,/static/dbb47454a74ffdac6ab079de768cc8af/4994d/logo-dark.png 300w" alt="EPI2ME Blog"/></picture></noscript><script type="module">const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1,e.parentNode.parentNode.querySelector("[data-placeholder-image]").style.opacity=0)}}</script></div></a></div><style data-emotion-css="14lrstn">.css-14lrstn{box-sizing:border-box;margin:0;min-width:0;-webkit-flex-basis:auto;-ms-flex-preferred-size:auto;flex-basis:auto;min-width:auto;-webkit-order:4;-ms-flex-order:4;order:4;}@media screen and (min-width:900px){.css-14lrstn{-webkit-flex-basis:66.666667%;-ms-flex-preferred-size:66.666667%;flex-basis:66.666667%;-webkit-order:unset;-ms-flex-order:unset;order:unset;}}</style><div class="css-14lrstn"><style data-emotion-css="aluv6h">.css-aluv6h{box-sizing:border-box;margin:0;min-width:0;display:none;min-width:auto;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}@media screen and (min-width:900px){.css-aluv6h{display:block;}}.css-aluv6h .button-group-link.level-2,.css-aluv6h .button-group-link.level-3{color:var(--theme-ui-colors-omegaDark,#e2e8f0);}.css-aluv6h .button-group-link.level-2:hover,.css-aluv6h .button-group-link.level-3:hover{color:var(--theme-ui-colors-alpha,#0084A9);}</style><div class="css-aluv6h"><style data-emotion-css="vurnku">.css-vurnku{box-sizing:border-box;margin:0;min-width:0;}</style><div class="css-vurnku"><style data-emotion-css="nsbjy5">.css-nsbjy5{box-sizing:border-box;margin:0;min-width:0;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:end;-webkit-justify-content:flex-end;-ms-flex-pack:end;justify-content:flex-end;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}</style><div class="css-nsbjy5"><style data-emotion-css="1yx0cqm">.css-1yx0cqm{box-sizing:border-box;margin:0;min-width:0;}.css-1yx0cqm + .css-1yx0cqm{margin-left:2rem;}</style><div class="css-1yx0cqm"><style data-emotion-css="1wekkny">.css-1wekkny{box-sizing:border-box;margin:0;min-width:0;}.css-1wekkny .button-group-link{padding-top:1rem;padding-bottom:1rem;}.css-1wekkny > * + a,.css-1wekkny > * + div,.css-1wekkny > * + button{margin-left:1rem;margin-top:2rem;}@media screen and (min-width:640px){.css-1wekkny > * + a,.css-1wekkny > * + div,.css-1wekkny > * + button{margin-top:0;}}@media screen and (min-width:1400px){.css-1wekkny > * + a,.css-1wekkny > * + div,.css-1wekkny > * + button{margin-left:2rem;}}</style><div class="css-1wekkny"><style data-emotion-css="1j4oxmh">.css-1j4oxmh{box-sizing:border-box;margin:0;min-width:0;position:relative;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}@media screen and (min-width:900px){.css-1j4oxmh{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}}.css-1j4oxmh:hover > .container-level-1{opacity:1;visibility:inherit;}@media screen and (min-width:900px){.css-1j4oxmh:hover > .container-level-1{-webkit-transform:translate(-50%,0);-ms-transform:translate(-50%,0);transform:translate(-50%,0);}}.css-1j4oxmh > svg{width:24px;height:24px;position:absolute;right:0;margin-left:auto;}@media screen and (min-width:640px){.css-1j4oxmh > svg{position:static;}}</style><div class="button-group-link level-1 css-1j4oxmh"><style data-emotion-css="12hibeq">.css-12hibeq{box-sizing:border-box;margin:0;min-width:0;display:inline-block;}</style><div class="css-12hibeq"> <!-- -->Resources</div><svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 24 24" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg"><path d="M16.293 9.293 12 13.586 7.707 9.293l-1.414 1.414L12 16.414l5.707-5.707z"></path></svg><style data-emotion-css="ua5l62">.css-ua5l62{box-sizing:border-box;margin:0;min-width:0;}.css-ua5l62.container-level-1{position:static;background-color:var(--theme-ui-colors-omegaLighter,#1a202c);display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-transition:all 250ms ease;transition:all 250ms ease;padding:1rem;margin-top:1rem;}@media screen and (min-width:900px){.css-ua5l62.container-level-1{position:absolute;top:100%;left:50%;-webkit-transform:translate(-50%,-20px);-ms-transform:translate(-50%,-20px);transform:translate(-50%,-20px);box-shadow:0 0 25px rgba(140,152,164,.355);visibility:hidden;overflow:hidden;opacity:0;background-color:var(--theme-ui-colors-contentBg,#2d3748);border-radius:0.5rem;padding:0;margin-top:0;}}.css-ua5l62.container-level-1.no-collection{padding:1rem;}@media screen and (min-width:900px){.css-ua5l62.container-level-1.no-collection{padding:1rem;}}.css-ua5l62.container-level-1.no-collection .button-group-link{padding-left:0.5rem;padding-right:0.5rem;padding-top:0.25rem;padding-bottom:0.25rem;margin-top:0.25rem;margin-bottom:0.25rem;}.css-ua5l62.container-level-2{-webkit-flex-basis:100%;-ms-flex-preferred-size:100%;flex-basis:100%;margin-top:0;}@media screen and (min-width:640px){.css-ua5l62.container-level-2{margin-top:0;}}@media screen and (min-width:900px){.css-ua5l62.container-level-2{margin-top:0.5rem;}}.css-ua5l62.container-level-2 .button-group-link{display:block;padding-top:0;padding-bottom:0;margin-top:0.5rem;}</style><div class="container-level-1  css-ua5l62"><style data-emotion-css="1fsj7ou">.css-1fsj7ou{box-sizing:border-box;margin:0;min-width:0;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;margin-top:1rem;background-color:var(--theme-ui-colors-omegaLighter,#1a202c);background:omegaLighter;}@media screen and (min-width:900px){.css-1fsj7ou{-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;padding:2rem;margin-top:0;}}.css-1fsj7ou > .button-group-link{-webkit-flex-basis:33.333333%;-ms-flex-preferred-size:33.333333%;flex-basis:33.333333%;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-wrap:wrap;-ms-flex-wrap:wrap;flex-wrap:wrap;-webkit-align-items:flex-start;-webkit-box-align:flex-start;-ms-flex-align:flex-start;align-items:flex-start;-webkit-align-content:flex-start;-ms-flex-line-pack:start;align-content:flex-start;min-width:210px;margin-bottom:0;padding-top:0;padding-bottom:0;}@media screen and (min-width:900px){.css-1fsj7ou > .button-group-link{margin-bottom:inherit;}}</style><div class="css-1fsj7ou"><div class="button-group-link level-2 css-1j4oxmh"><div class="css-12hibeq"> <!-- -->GETTING STARTED</div><div class="container-level-2 no-collection css-ua5l62"><style data-emotion-css="1sddnc9">.css-1sddnc9{box-sizing:border-box;margin:0;min-width:0;position:relative;}</style><a class="button-group-link level-3 css-1sddnc9" href="/downloads/"><div class="css-12hibeq"><style data-emotion-css="1gnm8id">.css-1gnm8id{box-sizing:border-box;margin:0;min-width:0;margin-right:0.25rem;display:inline-block;vertical-align:middle;border-radius:1rem;min-width:auto;}</style><div class="css-1gnm8id"><style data-emotion-css="1nxpe1r">.css-1nxpe1r{fill:var(--theme-ui-colors-omegaDarker,#a0aec0);width:24px;height:24px;display:block;}</style></div> <!-- -->Downloads</div></a><a class="button-group-link level-3 css-1sddnc9" href="/epi2me-docs/installation/"><div class="css-12hibeq"><div class="css-1gnm8id"></div> <!-- -->Installation</div></a><a class="button-group-link level-3 css-1sddnc9" href="/epi2me-docs/quickstart/"><div class="css-12hibeq"><div class="css-1gnm8id"></div> <!-- -->Quick Start</div></a><a class="button-group-link level-3 css-1sddnc9" href="/epi2me-docs/help/troubleshooting/"><div class="css-12hibeq"><div class="css-1gnm8id"></div> <!-- -->Troubleshooting</div></a></div></div><div class="button-group-link level-2 css-1j4oxmh"><div class="css-12hibeq"> <!-- -->RESOURCES</div><div class="container-level-2 no-collection css-ua5l62"><style data-emotion-css="ncew3f">.css-ncew3f{box-sizing:border-box;margin:0;min-width:0;width:200px;position:relative;}</style><a href="https://epi2me.nanoporetech.com/epi2me-docs/" class="button-group-link level-3 css-ncew3f"><div class="css-12hibeq"><div class="css-1gnm8id"></div> <!-- -->Documentation</div></a><a class="button-group-link level-3 css-1sddnc9" href="/wfindex/"><div class="css-12hibeq"><div class="css-1gnm8id"></div> <!-- -->Workflows</div></a><a class="button-group-link level-3 css-1sddnc9" href="/dataindex/"><div class="css-12hibeq"><div class="css-1gnm8id"></div> <!-- -->ONT Open Data</div></a><a class="button-group-link level-3 css-1sddnc9" href="/epi2me-docs/help/faq/"><div class="css-12hibeq"><div class="css-1gnm8id"></div> <!-- -->FAQs</div></a></div></div><div class="button-group-link level-2 css-1j4oxmh"><div class="css-12hibeq"> <!-- -->BLOG</div><div class="container-level-2 no-collection css-ua5l62"><a class="button-group-link level-3 css-1sddnc9" href="/category/articles/"><div class="css-12hibeq"><div class="css-1gnm8id"></div> <!-- -->Articles</div></a><a class="button-group-link level-3 css-1sddnc9" href="/category/how-tos/"><div class="css-12hibeq"><div class="css-1gnm8id"></div> <!-- -->How Tos</div></a><a class="button-group-link level-3 css-1sddnc9" href="/category/software-releases/"><div class="css-12hibeq"><div class="css-1gnm8id"></div> <!-- -->Software Releases</div></a><a class="button-group-link level-3 css-1sddnc9" href="/category/data-releases/"><div class="css-12hibeq"><div class="css-1gnm8id"></div> <!-- -->Data Releases</div></a></div></div></div></div></div><div class="button-group-link level-1 css-1j4oxmh"><div class="css-12hibeq"> <!-- -->About</div><svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 24 24" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg"><path d="M16.293 9.293 12 13.586 7.707 9.293l-1.414 1.414L12 16.414l5.707-5.707z"></path></svg><div class="container-level-1  css-ua5l62"><div class="css-1fsj7ou"><div class="button-group-link level-2 css-1j4oxmh"><div class="css-12hibeq"> <!-- -->ABOUT</div><div class="container-level-2 no-collection css-ua5l62"><a class="button-group-link level-3 css-1sddnc9" href="/about/"><div class="css-12hibeq"><div class="css-1gnm8id"></div> <!-- -->About</div></a><a class="button-group-link level-3 css-1sddnc9" href="/authors/"><div class="css-12hibeq"><div class="css-1gnm8id"></div> <!-- -->Our Team</div></a><a href="https://github.com/epi2me-labs" target="_blank" class="button-group-link level-3 css-1sddnc9"><div class="css-12hibeq"><div class="css-1gnm8id"></div> <!-- -->GitHub</div></a><a class="button-group-link level-3 css-1sddnc9" href="/epi2me-docs/help/contact-us/"><div class="css-12hibeq"><div class="css-1gnm8id"></div> <!-- -->Contact Us</div></a></div></div><div class="button-group-link level-2 css-1j4oxmh"><div class="css-12hibeq"> <!-- -->Related Information</div><div class="container-level-2 no-collection css-ua5l62"><a href="https://www.nanoporetech.com/community" class="button-group-link level-3 css-1sddnc9"><div class="css-12hibeq"><div class="css-1gnm8id"></div> <!-- -->Nanopore Community</div></a><a href="https://www.nanoporetech.com/privacy-policy" class="button-group-link level-3 css-1sddnc9"><div class="css-12hibeq"><div class="css-1gnm8id"></div> <!-- -->Privacy Policy</div></a><a href="https://www.nanoporetech.com/commercial_information" target="_blank" class="button-group-link level-3 css-1sddnc9"><div class="css-12hibeq"><div class="css-1gnm8id"></div> <!-- -->Terms of Use</div></a></div></div></div></div></div></div></div></div></div></div><style data-emotion-css="c60bxg">.css-c60bxg{box-sizing:border-box;margin:0;min-width:0;display:block;}@media screen and (min-width:900px){.css-c60bxg{display:none;}}</style><div class="css-c60bxg"><div class="css-vurnku"><style data-emotion-css="gh5cbz">.css-gh5cbz{box-sizing:border-box;margin:0;min-width:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;padding:0.25rem;width:32px;height:32px;color:inherit;background-color:transparent;border:none;border-radius:4px;display:inline-block;min-width:auto;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:space-evenly;-webkit-justify-content:space-evenly;-ms-flex-pack:space-evenly;justify-content:space-evenly;outline:0;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-tap-highlight-color:transparent;-webkit-touch-callout:none;-webkit-user-select:none;-khtml-user-select:none;-moz-user-select:none;-ms-user-select:none;border-radius:9999px;cursor:pointer;border-width:3px;border-style:solid;-webkit-transition:all 250ms ease;transition:all 250ms ease;box-sizing:initial;color:var(--theme-ui-colors-omega,#a0aec0);border-color:transparent;vertical-align:middle;width:24px;height:24px;padding:0.5rem;background-color:var(--theme-ui-colors-omegaLighter,#1a202c);vertical-align:middle;}.css-gh5cbz +a,.css-gh5cbz + button{margin-left:0.5rem;}.css-gh5cbz svg{width:24px;height:24px;}@media (hover:hover) and (pointer:fine){.css-gh5cbz:hover{background-color:var(--theme-ui-colors-alpha,#0084A9);color:var(--theme-ui-colors-white,#fff);}.css-gh5cbz:hover svg{fill:var(--theme-ui-colors-white,#fff);}}.css-gh5cbz:before{display:none;}.css-gh5cbz svg{width:32px;height:32px;}</style><button aria-label="Open Drawer" class="css-gh5cbz"><svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 448 512" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg"><path d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"></path></svg></button></div></div></div><style data-emotion-css="5g9gji">.css-5g9gji{box-sizing:border-box;margin:0;min-width:0;-webkit-flex-basis:auto;-ms-flex-preferred-size:auto;flex-basis:auto;min-width:auto;-webkit-order:3;-ms-flex-order:3;order:3;margin-left:1rem;margin-right:1rem;}@media screen and (min-width:900px){.css-5g9gji{-webkit-flex-basis:33.333333%;-ms-flex-preferred-size:33.333333%;flex-basis:33.333333%;-webkit-order:unset;-ms-flex-order:unset;order:unset;}}</style><div class="css-5g9gji"><style data-emotion-css="1t71mvf">.css-1t71mvf{box-sizing:border-box;margin:0;min-width:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;padding:0.25rem;width:32px;height:32px;color:inherit;background-color:transparent;border:none;border-radius:4px;display:inline-block;min-width:auto;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:space-evenly;-webkit-justify-content:space-evenly;-ms-flex-pack:space-evenly;justify-content:space-evenly;outline:0;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-tap-highlight-color:transparent;-webkit-touch-callout:none;-webkit-user-select:none;-khtml-user-select:none;-moz-user-select:none;-ms-user-select:none;border-radius:9999px;cursor:pointer;border-width:3px;border-style:solid;-webkit-transition:all 250ms ease;transition:all 250ms ease;box-sizing:initial;color:var(--theme-ui-colors-omega,#a0aec0);border-color:transparent;vertical-align:middle;width:24px;height:24px;padding:0.5rem;background-color:var(--theme-ui-colors-omegaLighter,#1a202c);display:block;}.css-1t71mvf +a,.css-1t71mvf + button{margin-left:0.5rem;}.css-1t71mvf svg{width:24px;height:24px;}@media (hover:hover) and (pointer:fine){.css-1t71mvf:hover{background-color:var(--theme-ui-colors-alpha,#0084A9);color:var(--theme-ui-colors-white,#fff);}.css-1t71mvf:hover svg{fill:var(--theme-ui-colors-white,#fff);}}.css-1t71mvf:before{display:none;}@media screen and (min-width:900px){.css-1t71mvf{display:none;}}</style><button aria-label="Search" class="css-1t71mvf"><svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 512 512" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg"><path d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"></path></svg></button><style data-emotion-css="f7lfdt">.css-f7lfdt{box-sizing:border-box;margin:0;min-width:0;display:none;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;background-color:var(--theme-ui-colors-omegaLight,#001A21);border-radius:0.5rem;position:static;top:2rem;left:50%;-webkit-transform:translate(0,0);-ms-transform:translate(0,0);transform:translate(0,0);z-index:99;width:auto;max-width:none;border-style:solid;border-color:var(--theme-ui-colors-headerBg,#1a202c);border-width:3px;padding-left:1rem;padding-right:1rem;}@media screen and (min-width:900px){.css-f7lfdt{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}}</style><div class="css-f7lfdt"><svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 512 512" style="flex-shrink:0" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg"><path d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"></path></svg><style data-emotion-css="dmqmfc">.css-dmqmfc{cursor:pointer;color:transparent;}</style><style data-emotion-css="1bbwzzs">.css-1bbwzzs{box-sizing:border-box;margin:0;min-width:0;display:block;width:100%;padding:0.5rem;-webkit-appearance:none;-moz-appearance:none;appearance:none;font-size:inherit;line-height:inherit;border:1px solid;border-radius:4px;color:inherit;background-color:transparent;font-family:'Inter',sans-serif;border:none;background-color:var(--theme-ui-colors-omegaLight,#001A21);color:var(--theme-ui-colors-omegaDark,#e2e8f0);outline:none;margin-left:0.25rem;cursor:pointer;color:transparent;}.css-1bbwzzs::-webkit-input-placeholder{color:var(--theme-ui-colors-omegaDark,#e2e8f0);}.css-1bbwzzs::-moz-placeholder{color:var(--theme-ui-colors-omegaDark,#e2e8f0);}.css-1bbwzzs:-ms-input-placeholder{color:var(--theme-ui-colors-omegaDark,#e2e8f0);}.css-1bbwzzs::placeholder{color:var(--theme-ui-colors-omegaDark,#e2e8f0);}</style><input type="text" placeholder="Search help and more..." aria-label="Search" class="css-1bbwzzs"/></div></div><style data-emotion-css="1ou1j4i">.css-1ou1j4i{box-sizing:border-box;margin:0;min-width:0;min-width:auto;-webkit-order:2;-ms-flex-order:2;order:2;}@media screen and (min-width:900px){.css-1ou1j4i{-webkit-order:unset;-ms-flex-order:unset;order:unset;}}</style><div class="css-1ou1j4i"><div class="css-vurnku"><button aria-label="Toggle dark mode" class="css-1t71mvf"><svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 512 512" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg"><path d="M283.211 512c78.962 0 151.079-35.925 198.857-94.792 7.068-8.708-.639-21.43-11.562-19.35-124.203 23.654-238.262-71.576-238.262-196.954 0-72.222 38.662-138.635 101.498-174.394 9.686-5.512 7.25-20.197-3.756-22.23A258.156 258.156 0 0 0 283.211 0c-141.309 0-256 114.511-256 256 0 141.309 114.511 256 256 256z"></path></svg></button><style data-emotion-css="gkja4h">.css-gkja4h{display:none;}@media screen and (min-width:900px){.css-gkja4h{display:block;}}.css-gkja4h.rc-switch{background:linear-gradient(45deg,#FF7D6B,#FFD64C);border:0;width:50px;height:24px;}.css-gkja4h.rc-switch:focus{box-shadow:none;}.css-gkja4h.rc-switch-checked{background:linear-gradient(45deg,#3366FF,#AC15B8);border:0;}.css-gkja4h.rc-switch:after{width:21px;height:21px;}.css-gkja4h.rc-switch-checked:after{left:auto;right:2px;}.css-gkja4h .rc-switch-inner{font-size:unset;top:1px;left:26px;}.css-gkja4h.rc-switch-checked .rc-switch-inner{left:7px;}</style><button aria-label="Toggle dark mode" type="button" role="switch" aria-checked="true" class="rc-switch css-gkja4h rc-switch-checked"><span class="rc-switch-inner"><span class="rc-switch-inner-checked"><style data-emotion-css="uk6cul">.css-uk6cul{vertical-align:middle;}</style><svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 512 512" class="css-uk6cul" height="16" width="16" xmlns="http://www.w3.org/2000/svg"><path d="M283.211 512c78.962 0 151.079-35.925 198.857-94.792 7.068-8.708-.639-21.43-11.562-19.35-124.203 23.654-238.262-71.576-238.262-196.954 0-72.222 38.662-138.635 101.498-174.394 9.686-5.512 7.25-20.197-3.756-22.23A258.156 258.156 0 0 0 283.211 0c-141.309 0-256 114.511-256 256 0 141.309 114.511 256 256 256z"></path></svg></span><span class="rc-switch-inner-unchecked"><svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 512 512" class="css-uk6cul" height="16" width="16" xmlns="http://www.w3.org/2000/svg"><path d="M256 160c-52.9 0-96 43.1-96 96s43.1 96 96 96 96-43.1 96-96-43.1-96-96-96zm246.4 80.5l-94.7-47.3 33.5-100.4c4.5-13.6-8.4-26.5-21.9-21.9l-100.4 33.5-47.4-94.8c-6.4-12.8-24.6-12.8-31 0l-47.3 94.7L92.7 70.8c-13.6-4.5-26.5 8.4-21.9 21.9l33.5 100.4-94.7 47.4c-12.8 6.4-12.8 24.6 0 31l94.7 47.3-33.5 100.5c-4.5 13.6 8.4 26.5 21.9 21.9l100.4-33.5 47.3 94.7c6.4 12.8 24.6 12.8 31 0l47.3-94.7 100.4 33.5c13.6 4.5 26.5-8.4 21.9-21.9l-33.5-100.4 94.7-47.3c13-6.5 13-24.7.2-31.1zm-155.9 106c-49.9 49.9-131.1 49.9-181 0-49.9-49.9-49.9-131.1 0-181 49.9-49.9 131.1-49.9 181 0 49.9 49.9 49.9 131.1 0 181z"></path></svg></span></span></button></div></div></div></div></div><style data-emotion-css="1qe1pxe">.css-1qe1pxe{box-sizing:border-box;margin:0;min-width:0;-webkit-flex:1 1 auto;-ms-flex:1 1 auto;flex:1 1 auto;}</style><div class="css-1qe1pxe"><style data-emotion-css="14hbchx">.css-14hbchx{box-sizing:border-box;margin:0;min-width:0;width:100%;max-width:1400px;margin-left:auto;margin-right:auto;max-width:1400px;padding-left:1rem;padding-right:1rem;}@media screen and (min-width:640px){.css-14hbchx{padding-left:2rem;padding-right:2rem;}}</style><div class="css-14hbchx"><style data-emotion-css="1teua8m">.css-1teua8m{box-sizing:border-box;margin:0;min-width:0;width:100%;max-width:1400px;margin-left:auto;margin-right:auto;max-width:none;}</style><div class="css-1teua8m"><style data-emotion-css="4cffwv">.css-4cffwv{box-sizing:border-box;margin:0;min-width:0;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}</style><div class="css-4cffwv"><div><style data-emotion-css="1syn9jw">.css-1syn9jw{box-sizing:border-box;margin:0;min-width:0;font-family:'Inter',sans-serif;font-weight:bold;line-height:1.33;display:block;color:var(--theme-ui-colors-heading,#edf2f7);font-weight:900;-webkit-text-decoration:none;text-decoration:none;margin-bottom:1rem;font-size:1.875rem;}@media screen and (min-width:640px){.css-1syn9jw{font-size:2.25rem;}}@media screen and (min-width:1400px){.css-1syn9jw{font-size:3rem;}}</style><h1 class="css-1syn9jw">Simulating sequencing datasets<!-- --> </h1><style data-emotion-css="5eg2lm">.css-5eg2lm{box-sizing:border-box;margin:0;min-width:0;display:block;color:var(--theme-ui-colors-heading,#edf2f7);font-weight:600;-webkit-text-decoration:none;text-decoration:none;margin-bottom:1rem;font-size:1rem;font-weight:normal;color:var(--theme-ui-colors-omegaDark,#e2e8f0);margin-bottom:0;}</style><div class="css-5eg2lm"><style data-emotion-css="dwcav6">.css-dwcav6{box-sizing:border-box;margin:0;min-width:0;}.css-dwcav6 > *:not(:last-child) + *:before{content:" ・ ";}</style><div class="css-dwcav6"><div class="css-12hibeq">By <style data-emotion-css="1jns145">.css-1jns145{box-sizing:border-box;margin:0;min-width:0;color:var(--theme-ui-colors-omegaDark,#e2e8f0);-webkit-text-decoration:none;text-decoration:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;}.css-1jns145:visited{color:var(--theme-ui-colors-omegaDark,#e2e8f0);}.css-1jns145:hover{color:var(--theme-ui-colors-alpha,#0084A9);}</style><a class="css-1jns145" href="/author/georgette-tanner/"><strong>Georgette Tanner</strong></a></div><div class="css-12hibeq">Published in <a class="css-1jns145" href="/category/how-tos/"><strong>How Tos</strong></a></div><div class="css-12hibeq">March 18, 2024</div><div class="css-12hibeq"><strong>12<!-- --> min read</strong></div></div></div></div></div></div></div><style data-emotion-css="8s19zc">.css-8s19zc{box-sizing:border-box;margin:0;min-width:0;min-width:auto;border-top-style:solid;border-top-color:transparent;border-top-width:2px;height:0;margin-top:0.5rem;margin-bottom:0.5rem;}@media screen and (min-width:640px){.css-8s19zc{margin-top:1rem;margin-bottom:1rem;}}</style><div class="css-8s19zc"></div><div class="css-14hbchx"><div class="css-1teua8m"><div class="css-4cffwv"><style data-emotion-css="1nu4rf6">.css-1nu4rf6{box-sizing:border-box;margin:0;min-width:0;width:100%;max-width:1400px;margin-left:auto;margin-right:auto;-webkit-flex:1;-ms-flex:1;flex:1;margin-left:0;}</style><div class="css-1nu4rf6"><style data-emotion-css="15qiiq2">.css-15qiiq2{box-sizing:border-box;margin:0;min-width:0;background-color:var(--theme-ui-colors-contentBg,#2d3748);border-radius:1rem;-webkit-transition: -webkit-transform 250ms ease, box-shadow 250ms ease, color 250ms ease;-webkit-transition: transform 250ms ease, box-shadow 250ms ease, color 250ms ease;transition: transform 250ms ease, box-shadow 250ms ease, color 250ms ease;box-shadow:1px 1px 5px 0 rgba(1,1,1,.05);padding:2rem;}</style><div class="css-15qiiq2"><style data-emotion-css="1xptjhn">.css-1xptjhn{min-height:23rem;margin-top:-2rem;margin-left:-2rem;margin-right:-2rem;}.css-1xptjhn img{border-radius:1rem 1rem 0 0;}</style><div data-gatsby-image-wrapper="" class="gatsby-image-wrapper gatsby-image-wrapper-constrained css-1xptjhn"><div style="max-width:1600px;display:block"><img alt="" role="presentation" aria-hidden="true" src="data:image/svg+xml;charset=utf-8,%3Csvg%20height=&#x27;649.6&#x27;%20width=&#x27;1600&#x27;%20xmlns=&#x27;http://www.w3.org/2000/svg&#x27;%20version=&#x27;1.1&#x27;%3E%3C/svg%3E" style="max-width:100%;display:block;position:static"/></div><div aria-hidden="true" data-placeholder-image="" style="opacity:1;transition:opacity 500ms linear;background-color:#c8c8d8;position:absolute;top:0;left:0;bottom:0;right:0"></div><picture><source type="image/webp" data-srcset="/static/676ccc6f1894f739f61cc237322c970e/0347d/polar_bear.webp 750w,/static/676ccc6f1894f739f61cc237322c970e/b4d1b/polar_bear.webp 1500w" sizes="(min-width: 1500px) 1500px, 100vw"/><img data-gatsby-image-ssr="" data-main-image="" style="opacity:0" sizes="(min-width: 1500px) 1500px, 100vw" decoding="async" loading="lazy" data-src="/static/676ccc6f1894f739f61cc237322c970e/23364/polar_bear.jpg" data-srcset="/static/676ccc6f1894f739f61cc237322c970e/c8f6b/polar_bear.jpg 750w,/static/676ccc6f1894f739f61cc237322c970e/23364/polar_bear.jpg 1500w" alt="Simulating sequencing datasets"/></picture><noscript><picture><source type="image/webp" srcSet="/static/676ccc6f1894f739f61cc237322c970e/0347d/polar_bear.webp 750w,/static/676ccc6f1894f739f61cc237322c970e/b4d1b/polar_bear.webp 1500w" sizes="(min-width: 1500px) 1500px, 100vw"/><img data-gatsby-image-ssr="" data-main-image="" style="opacity:0" sizes="(min-width: 1500px) 1500px, 100vw" decoding="async" loading="lazy" src="/static/676ccc6f1894f739f61cc237322c970e/23364/polar_bear.jpg" srcSet="/static/676ccc6f1894f739f61cc237322c970e/c8f6b/polar_bear.jpg 750w,/static/676ccc6f1894f739f61cc237322c970e/23364/polar_bear.jpg 1500w" alt="Simulating sequencing datasets"/></picture></noscript><script type="module">const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1,e.parentNode.parentNode.querySelector("[data-placeholder-image]").style.opacity=0)}}</script></div><div class="css-8s19zc"></div><style data-emotion-css="p9c5cf">.css-p9c5cf{padding:0;margin:0;margin-bottom:2rem;color:var(--theme-ui-colors-article,#e2e8f0);font-size:1.125rem;padding:0;margin:0;margin-bottom:2rem;color:var(--theme-ui-colors-article,#e2e8f0);font-size:1.125rem;}.css-p9c5cf code{font-size:inherit;}.css-p9c5cf code{font-size:inherit;}</style><p class="css-p9c5cf">Bioinformatics tools and pipelines, as with any analysis methods, need to be tested to ensure they work as expected.
This requires benchmarking them to determine their accuracy and to assess the impact of various factors on the results.
Such a task requires data for which the ground truth is known, i.e. knowing what a 100% accurate result should look like.
While there are some situations where users may be able to rely on real sequencing data from robustly characterised mock samples in the lab, simulating datasets on a computer is commonly the only practical and reliable option.</p><p class="css-p9c5cf">But how do you go about simulating datasets fit for this purpose?
This is a huge field of research, as simulating realistic sequencing data is not trivial.
The goal is to create a set of sequencing reads that reflect not only the biology of an underlying hypothetical sample, but also the patterns and noise associated with real sequencing data.
Such variation results from a variety of sources, including:</p><style data-emotion-css="16v5hhi">.css-16v5hhi{padding:0;margin:0;margin-bottom:2rem;list-style-position:outside;list-style-image:none;margin-left:1rem;padding:0;margin:0;margin-bottom:2rem;list-style-position:outside;list-style-image:none;margin-left:1rem;}</style><ul class="css-16v5hhi"><style data-emotion-css="1o70w8g">.css-1o70w8g{margin-bottom:0.5rem;padding-left:0;color:var(--theme-ui-colors-article,#e2e8f0);font-size:1.125rem;margin:0;margin-bottom:0.5rem;padding-left:0;color:var(--theme-ui-colors-article,#e2e8f0);font-size:1.125rem;margin:0;}.css-1o70w8g ol{margin-top:0.5rem;margin-bottom:0.5rem;margin-left:1rem;}.css-1o70w8g ul{margin-top:0.5rem;margin-bottom:0.5rem;margin-left:1rem;}.css-1o70w8g p{margin-bottom:0.5rem;}.css-1o70w8g code{font-size:inherit;}.css-1o70w8g ol{margin-top:0.5rem;margin-bottom:0.5rem;margin-left:1rem;}.css-1o70w8g ul{margin-top:0.5rem;margin-bottom:0.5rem;margin-left:1rem;}.css-1o70w8g p{margin-bottom:0.5rem;}.css-1o70w8g code{font-size:inherit;}</style><li class="css-1o70w8g">library preparation - preparation introduces several sources of variation. Fragmentation of molecules, errors and biases introduced by PCR amplification, formation of chimeric molecules, and biases introduced through ligation or hybridization of molecules to artificial constructs or probes, all contribute to the complexity of the resulting dataset.</li><li class="css-1o70w8g">sequencing - the sequencing process itself, including the sequencing platform and basecaller, can introduce both systematic and random error into the recorded base sequences.</li></ul><p class="css-p9c5cf">Having unrealistic data can prevent you from getting accurate benchmarking results, and so your choice of methods for simulating data needs to be considered carefully.
But don’t let this put you off!
This guide will help you to understand the challenges associated with simulating sequencing datasets, so that you can avoid potential pitfalls and gain reliable benchmarking results.</p><p class="css-p9c5cf">Numerous genome simulation tools exist to automate many of the steps involved in simulating sequencing datasets, with new ones frequently becoming available and each tailored to different scenarios. </p><p class="css-p9c5cf">While we won’t delve into the full range of simulation tools here (due to their sheer abundance and variety), we’ll cover the general approaches employed by the different methods and mention a few examples towards the end.</p><style data-emotion-css="1kspjju">.css-1kspjju{padding:0;margin:0;margin-bottom:1rem;font-size:1.5rem;font-family:'Inter',sans-serif;line-height:1.33;font-weight:bold;display:block;color:var(--theme-ui-colors-heading,#edf2f7);-webkit-text-decoration:none;text-decoration:none;}</style><style data-emotion-css="1ophxj7">.css-1ophxj7{pointer-events:painted;padding:0;margin:0;margin-bottom:1rem;font-size:1.5rem;font-family:'Inter',sans-serif;line-height:1.33;font-weight:bold;display:block;color:var(--theme-ui-colors-heading,#edf2f7);-webkit-text-decoration:none;text-decoration:none;}.css-1ophxj7 svg{visibility:hidden;width:24px;height:24px;position:absolute;top:50%;-webkit-transform:translate(0,-50%);-ms-transform:translate(0,-50%);transform:translate(0,-50%);padding-left:0.5rem;}.css-1ophxj7:hover svg{visibility:visible;}</style><h2 id="general-approaches" class="css-1ophxj7"><style data-emotion-css="3n7dj1">.css-3n7dj1{box-sizing:border-box;margin:0;min-width:0;display:block;color:var(--theme-ui-colors-heading,#edf2f7);font-weight:bold;-webkit-text-decoration:none;text-decoration:none;margin-bottom:1rem;font-size:1.5rem;position:relative;}</style><a href="#general-approaches" class="css-3n7dj1">General approaches<svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 512 512" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg"><path d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"></path></svg></a></h2><p class="css-p9c5cf">The most generic, or at least brute force, approach to simulating sequencing data would be start from known reference sequences and fully replicate biological and experimental variation.
Such a method gives precise knowledge and traceability of all the reads that come out of the simulation.
However this method naturally requires exquisite knowledge of physical processes in the laboratory and the sequencing system, so can be hard to accomplish.</p><p class="css-p9c5cf">An alternative approach is to manipulate real sequencing reads in a controlled fashion.
This has the benefit of not needing to characterise explicitely the complete experimental setup, but does require having adequate knowledge of precisely what was sequenced in the first instance.</p><p class="css-p9c5cf">We discuss both methods below.</p><style data-emotion-css="1mhehvz">.css-1mhehvz{padding:0;margin:0;margin-bottom:1rem;font-size:1.125rem;font-family:'Inter',sans-serif;line-height:1.33;font-weight:bold;display:block;color:var(--theme-ui-colors-heading,#edf2f7);-webkit-text-decoration:none;text-decoration:none;}.css-1mhehvz + pre{margin-top:0;}</style><style data-emotion-css="11m2vzb">.css-11m2vzb{pointer-events:painted;padding:0;margin:0;margin-bottom:1rem;font-size:1.125rem;font-family:'Inter',sans-serif;line-height:1.33;font-weight:bold;display:block;color:var(--theme-ui-colors-heading,#edf2f7);-webkit-text-decoration:none;text-decoration:none;}.css-11m2vzb svg{visibility:hidden;width:24px;height:24px;position:absolute;top:50%;-webkit-transform:translate(0,-50%);-ms-transform:translate(0,-50%);transform:translate(0,-50%);padding-left:0.5rem;}.css-11m2vzb:hover svg{visibility:visible;}.css-11m2vzb + pre{margin-top:0;}</style><h3 id="simulation-from-reference-sequences" class="css-11m2vzb"><style data-emotion-css="3mxrie">.css-3mxrie{box-sizing:border-box;margin:0;min-width:0;display:block;color:var(--theme-ui-colors-heading,#edf2f7);font-weight:bold;-webkit-text-decoration:none;text-decoration:none;margin-bottom:1rem;font-size:1.125rem;position:relative;}</style><a href="#simulation-from-reference-sequences" class="css-3mxrie">Simulation from reference sequences<svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 512 512" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg"><path d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"></path></svg></a></h3><p class="css-p9c5cf">Simulating sequencing reads starting from a small collection of reference sequences can be described as a “generative approach”.
It aims to create simulated reads by applying a chain of (probabilistic) logical transforms which aim to reflect reality. </p><p class="css-p9c5cf">The first step is to simulate biological variation that might be present in a sample, that is not present in our collection of reference sequences.
We can introduce mutations in the reference sequences, that are then used in subsequent steps.
For example, we could take FASTA file containing a genome sequence and swap the base at position 100 from a T to a G to simulate a point mutation.
Or we could copy the sequence from position 50-59 and insert it between bases 59 and 60, to simulate a copy number variant at position 50-59 with a copy number of 2.</p><p class="css-p9c5cf">One of the challenges with modifying a genome sequence is keeping track of where each base corresponds to in the original sequence.
As soon as any insertions or deletions occur in a sequence, the Nth base in the original sequence may not match up with the Nth base in the modified sequence, and so consideration needs to be given into working out where any subsequent variants should be placed.
Various approaches are used to handle this.
One option is to incorporate variants in order from last to first so that the sequence prior to the last incorporated variant position always corresponds to the original sequence.
This works well when the larger insertions don’t overlap other variants, but in situations where they do, more sophisticated methods are required.
Another option is to use a “blueprint” representation of the sequence, that can more easily be manipulated during the simulation.
This blueprint takes the form of a list of sections, each representing either a portion of the original sequence, in which case only a start and end position is recorded, or instead a variant sequence.
The list of sections is sequentially modified to represent each new variant, and once all the variants are incorporated, it can be used as instructions for building the final simulated sequence.
For each section either the relevant portion of the original sequence or, if present, the recorded variant sequence is copied and concatenated together to form the full simulated sequence.</p><p class="css-p9c5cf"><style data-emotion-css="190buoc">.css-190buoc{box-shadow:1px 1px 5px 0 rgba(1,1,1,.15);overflow:hidden;background-color:var(--theme-ui-colors-contentBg,#2d3748);border-radius:0.5rem;margin:0;}</style><style data-emotion-css="cm98ls">.css-cm98ls{box-shadow:1px 1px 5px 0 rgba(1,1,1,.15);overflow:hidden;background-color:var(--theme-ui-colors-contentBg,#2d3748);border-radius:0.5rem;margin:0;box-shadow:1px 1px 5px 0 rgba(1,1,1,.15);overflow:hidden;background-color:var(--theme-ui-colors-contentBg,#2d3748);border-radius:0.5rem;margin:0;}</style><figure class="gatsby-resp-image-figure css-cm98ls">
    <span class="gatsby-resp-image-wrapper" style="position:relative;display:block;margin-left:auto;margin-right:auto;max-width:1140px">
      <span class="gatsby-resp-image-background-image" style="padding-bottom:39.64912280701754%;position:relative;bottom:0;left:0;display:block"></span>
  <style data-emotion-css="1nh7vc8">.css-1nh7vc8{padding:0;margin:0;margin-bottom:1rem;max-width:100%;padding:0;margin:0;margin-bottom:1rem;max-width:100%;}</style><img class="gatsby-resp-image-image css-1nh7vc8" alt="Blueprint approach" title="An illustration of the blueprint approach for genome simulation." src="/static/c44b95d93e791c0e98246e37737d9333/b5cea/blueprint.png" srcSet="/static/c44b95d93e791c0e98246e37737d9333/0e2fe/blueprint.png 285w,/static/c44b95d93e791c0e98246e37737d9333/432e7/blueprint.png 570w,/static/c44b95d93e791c0e98246e37737d9333/b5cea/blueprint.png 1140w,/static/c44b95d93e791c0e98246e37737d9333/09ede/blueprint.png 1710w,/static/c44b95d93e791c0e98246e37737d9333/d50e7/blueprint.png 2280w,/static/c44b95d93e791c0e98246e37737d9333/4fe3b/blueprint.png 2901w" sizes="(max-width: 1140px) 100vw, 1140px" style="width:100%;height:100%;margin:0;vertical-align:middle;position:absolute;top:0;left:0" loading="lazy" decoding="async"/>
    </span>
    <style data-emotion-css="16wbd3m">.css-16wbd3m{text-align:center;font-style:italic;font-size:0.875rem;margin:1rem;}</style><style data-emotion-css="13jeo2h">.css-13jeo2h{text-align:center;font-style:italic;font-size:0.875rem;margin:1rem;text-align:center;font-style:italic;font-size:0.875rem;margin:1rem;}</style><figcaption class="gatsby-resp-image-figcaption css-13jeo2h">An illustration of the blueprint approach for genome simulation.</figcaption>
  </figure></p><p class="css-p9c5cf">Once we have our modified reference sequence, we next need to generate reads from it.
The aim to mimic real library preparation and the sequencing process by essentially selecting regions from the simulated genome sequences, using specified location and length distributions.
However, achieving this in a realistic manor is a significant challenge.
There are at least two important considerations, elucidated below.</p><p class="css-p9c5cf"><style data-emotion-css="1ukrlb9">.css-1ukrlb9{font-weight:700;font-weight:700;}</style><strong class="css-1ukrlb9"><em class="css-0">Read coverage</em></strong><br/>
Almost all library preparation and sequencing methods result in at least some degree of non-uniform read coverage along a genome.
Further biases may also manifest in the direction of reads, with a tendency for more reads to originate from either the forward or reverse strand.
In short-read sequencing, GC composition is a strong factor in how efficiently a region is sequenced, with many short read simulators taking this into account.
Further consideration needs to be taken when simulating amplicon, whole-exome, or other targeted sequencing datasets.
While simply limiting the reference sequence sampling to the relevant regions of the genome may be sufficient in some experiments being performed with simulated data, more tailored approaches are needed to reflect the nuances of these datasets.</p><p class="css-p9c5cf"><strong class="css-1ukrlb9"><em class="css-0">Incorporating sequencing error</em></strong><br/>
Real-world sequencing data contains various errors; if it didn’t then variant calling would be a lot easier!
These errors need to be incorporated into the simulated reads, with base quality scores that reflect the uncertainty.
At a basic level this might, for example, involve measuring empirical errors and quality distributions and applying these to the “perfect” reads derived from the previous step.
Such models can be made increasingly sophisticated and incorporate models of the surrounding sequence and read position context when simulating quality scores and errors.
For example in short-read sequencing by synthesis, a T nucleotide following GG is more likely to be incorrectly sequenced than a T following AA (<style data-emotion-css="14yhzo">.css-14yhzo{color:var(--theme-ui-colors-alpha,#0084A9);-webkit-text-decoration:none;text-decoration:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;color:var(--theme-ui-colors-alpha,#0084A9);-webkit-text-decoration:none;text-decoration:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;}.css-14yhzo:visited{color:var(--theme-ui-colors-alpha,#0084A9);}.css-14yhzo:hover{color:var(--theme-ui-colors-alphaDark,#a3bffa);}.css-14yhzo:visited{color:var(--theme-ui-colors-alpha,#0084A9);}.css-14yhzo:hover{color:var(--theme-ui-colors-alphaDark,#a3bffa);}</style><a href="https://doi.org/10.1093/nargab/lqab019" class="css-14yhzo">Stoler and Nekrutenko 2021</a>).
Similarly positions closer to the ends of reads from such sequencing methods can also have an increased error likelihood.</p><p class="css-p9c5cf">Taking the idea of the generative approach to the extreme one might instead physically model the measurement process and apply the same data processing as would ordinarily be applied to real data.
Such an approach was previously used within <a href="https://github.com/nanoporetech/scrappie" class="css-14yhzo">scrappie</a>, where a MinION “squiggles” (ionic current traces) could be simulated from a provided nucleotide sequenced, additional noise and variation added, with the resulting signal passed to a basecaller to produce a simulated basecall.</p><p class="css-p9c5cf">Many such simulation methods allow the user to train their own models on real data in order to account for the specific chemistry, library preparation, and genome context relevant to their study.
Even so, the reads generated by these methods can vary in how realistically they reflect real data. </p><p class="css-p9c5cf">The complexity in accurately modelling physical processes is the main rationale for using alternative simulation approaches which utilise and modify real sequencing datasets. </p><h3 id="simulation-from-real-data" class="css-11m2vzb"><a href="#simulation-from-real-data" class="css-3mxrie">Simulation from real-data<svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 512 512" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg"><path d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"></path></svg></a></h3><p class="css-p9c5cf">As a somewhat simpler approach, an alternative strategy for simulating read sets is to sample reads from existing experimental data.
This does not require any in-depth specialist knowlege and so can more easily achieve the desired datasets.</p><p class="css-p9c5cf">The basic idea is to subsample reads from an existing biological sample and merge them together with reads from other samples at required frequencies.
This can generate datasets with a specified coverage, and with variants, species, or lineages at specified frequencies.</p><p class="css-p9c5cf">The process isn’t without its disadvantages however.
Firstly flexibility is limited.
The simulation is restricted by the coverage, variants, species that are available in existing datasets.</p><p class="css-p9c5cf">Secondly, the technique requires thorough characterisation of the initial datasets with respect to the type of benchmarking planned.
If you want to test variant calling at different allele frequencies, then you need to know the positions of all true variants in the initial datasets, and if you want to test lineage analysis, then you need to be confident of the lineage composition of the initial datasets.
If this characterisation is not accurate, then any benchmarking performed on samples created from them will be unreliable.</p><p class="css-p9c5cf">In order to combat the first of these disadvantages, a modified method is to introduce mutations into the library of sequenced reads.
This approach avoids many of the issues associated with artificially generating reads <em class="css-0">de novo</em>, though it comes with some difficulties of its own.</p><p class="css-p9c5cf">The process relies on having accurate initial identification of the reads which are mutated: if we cannot confidently identify the reads then we will be introducing mutations other than those which we thought we were creating.
The identity of reads is typically made through alignment, and introduction of variants is through known genomic co-ordinates.
The mis-indentification (and therefore application of variants) can therefore occur due to ambiguities in the read alignment process.</p><p class="css-p9c5cf">The ambiguous alignment of reads is a whole topic of conversation in itself; let us discuss briefly consider one such example.
Consider the case where a genome contains non-identical repeats.
When aligning reads to this genome it can be the case that there is an ambiguity as to which repeat the read is derived.
There can be errors and biases in this assignment; particularly if sequencing errors conflate with the differences between the repeats.
If we apply variants to all reads which align to one copy of a repetitive element, we may in fact be applying variants to reads whose true identity is a second copy of the element.</p><p class="css-p9c5cf">Having introduced variants, this misalignment effect can work the other way round.
The modified reads may be significantly different from the original seed read, and may now more properly align to another locus of the genome entirely; however our synthetic dataset has the read aligned at the original location.
In a real sequencing dataset with the variant present, a read may align to the alternative location: our simulated read set does not now reflect a real set od aligned reads.
If we were to now benchmark a variant caller with the synthetic set we may obtain better results than with real reads (natively containing the variants).
Realigning the synthetic reads following artificial introduction of variants would better model a real experiment, and the resulting inaccuracies to be accounted for in subsequent benchmarking.</p><p class="css-p9c5cf">Finally, care should be taken when incorporating variants to ensure that other properties of the data are maintained.
For example: base qualities.
Suppose we insert two Gs before an existing pair of Gs.
In real data this would often result in low quality scores for the whole stretch of Gs due to it creating a 4-base homopolymer.
Similarly if a GGT trimer has an A inserted before the T, then in real data this should result in an increased quality score for the T. </p><h3 id="how-to-get-hold-of-well-characterised-real-datasets" class="css-11m2vzb"><a href="#how-to-get-hold-of-well-characterised-real-datasets" class="css-3mxrie">How to get hold of well characterised real datasets<svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 512 512" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg"><path d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"></path></svg></a></h3><p class="css-p9c5cf">So those are two different approaches used to simulate sequencing datasets.
You might now be asking, “how do I get hold of some thoroughly characterised real datasets?”</p><p class="css-p9c5cf">The best way to get a well characterised dataset again depends on your specific use case.
Previously characterised datasets are often available for common scenarios so it’s worth searching for one, or having a look at what’s been used in other simulation studies in your field.
For human data, consortia exist that provide extensively characterised samples, such as the <a href="https://www.nist.gov/programs-projects/genome-bottle" class="css-14yhzo">Genome in a Bottle Consortium</a>.
Among other methods, the GIAB consortia employ son/father/mother trios to determine sets of high confidence variant calls.
Other options may be to generate data yourself in such a way that you are confident of the ground truths.
For example, sequencing pure isolates of species for use in simulating datasets for abundance testing.</p><h2 id="which-approach-do-i-use" class="css-1ophxj7"><a href="#which-approach-do-i-use" class="css-3n7dj1">Which approach do I use?<svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 512 512" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg"><path d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"></path></svg></a></h2><h3 id="i-need-to-test-how-my-pipeline-performs-with-lower-coverage" class="css-11m2vzb"><a href="#i-need-to-test-how-my-pipeline-performs-with-lower-coverage" class="css-3mxrie">I need to test how my pipeline performs with lower coverage<svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 512 512" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg"><path d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"></path></svg></a></h3><p class="css-p9c5cf">Simple! Use the second approach by subsampling to reduce coverage of an existing dataset, either from FASTQ or BAM files:</p><style data-emotion-css="duxjoh">.css-duxjoh{padding:1rem;margin:0;margin-bottom:2rem;font-size:85%;color:#d6deeb;background-color:#011627;font-family:Consolas,Menlo,Monaco,source-code-pro,Courier New,monospace;-webkit-hyphens:none;-moz-hyphens:none;-ms-hyphens:none;hyphens:none;overflow:auto;border-radius:0.5rem;tab-size:4;margin-top:2rem;}.css-duxjoh .changed{color:rgb(162,191,252);font-style:italic;}.css-duxjoh .deleted{color:rgba(239,83,80,0.56);font-style:italic;}.css-duxjoh .inserted,.css-duxjoh .attr-name{color:rgb(173,219,103);font-style:italic;}.css-duxjoh .comment{color:rgb(99,119,119);font-style:italic;}.css-duxjoh .string,.css-duxjoh .url{color:rgb(173,219,103);}.css-duxjoh .variable{color:rgb(214,222,235);}.css-duxjoh .number{color:rgb(247,140,108);}.css-duxjoh .builtin,.css-duxjoh .char,.css-duxjoh .constant,.css-duxjoh .function{color:rgb(130,170,255);}.css-duxjoh .punctuation{color:rgb(199,146,234);}.css-duxjoh .selector,.css-duxjoh .doctype{color:rgb(199,146,234);font-style:italic;}.css-duxjoh .class-name{color:rgb(255,203,139);}.css-duxjoh .tag,.css-duxjoh .operator,.css-duxjoh .keyword{color:rgb(127,219,202);}.css-duxjoh .boolean{color:rgb(255,88,116);}.css-duxjoh .property{color:rgb(128,203,196);}.css-duxjoh .namespace{color:rgb(178,204,214);}</style><style data-emotion-css="12neuit">.css-12neuit{font-size:85%;}</style><!--$--><style data-emotion-css="dv20tu">.css-dv20tu{padding:1rem;margin:0;margin-bottom:2rem;font-size:85%;color:#d6deeb;background-color:#011627;font-family:Consolas,Menlo,Monaco,source-code-pro,Courier New,monospace;-webkit-hyphens:none;-moz-hyphens:none;-ms-hyphens:none;hyphens:none;overflow:auto;border-radius:0.5rem;tab-size:4;margin-top:2rem;font-size:85%;}.css-dv20tu .changed{color:rgb(162,191,252);font-style:italic;}.css-dv20tu .deleted{color:rgba(239,83,80,0.56);font-style:italic;}.css-dv20tu .inserted,.css-dv20tu .attr-name{color:rgb(173,219,103);font-style:italic;}.css-dv20tu .comment{color:rgb(99,119,119);font-style:italic;}.css-dv20tu .string,.css-dv20tu .url{color:rgb(173,219,103);}.css-dv20tu .variable{color:rgb(214,222,235);}.css-dv20tu .number{color:rgb(247,140,108);}.css-dv20tu .builtin,.css-dv20tu .char,.css-dv20tu .constant,.css-dv20tu .function{color:rgb(130,170,255);}.css-dv20tu .punctuation{color:rgb(199,146,234);}.css-dv20tu .selector,.css-dv20tu .doctype{color:rgb(199,146,234);font-style:italic;}.css-dv20tu .class-name{color:rgb(255,203,139);}.css-dv20tu .tag,.css-dv20tu .operator,.css-dv20tu .keyword{color:rgb(127,219,202);}.css-dv20tu .boolean{color:rgb(255,88,116);}.css-dv20tu .property{color:rgb(128,203,196);}.css-dv20tu .namespace{color:rgb(178,204,214);}</style><pre class="prism-code language-css-12neuit css-dv20tu"><div class="token-line"><style data-emotion-css="1baulvz">.css-1baulvz{display:inline-block;}</style><span class="token plain css-1baulvz">samtools view -bh -s {DOWNSAMPLE_FRACTION} {INPUT BAM} &gt; {SUBSAMPLED_BAM}</span></div></pre><!--/$--><p class="css-p9c5cf">Okay, that was a fairly obvious one to start with, but it does get more interesting!</p><h3 id="i-need-to-test-my-metagenomics-abundance-estimation-pipeline" class="css-11m2vzb"><a href="#i-need-to-test-my-metagenomics-abundance-estimation-pipeline" class="css-3mxrie">I need to test my metagenomics abundance estimation pipeline<svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 512 512" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg"><path d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"></path></svg></a></h3><p class="css-p9c5cf">The best option for this would likely be the second approach above, providing you have real datasets available for a sufficient variety of species, and importantly, of sufficient purity.
In metagenomic datasets we would hope that the relative abundance of reads should reflect the abundance of species in a sample.
However in reality, factors such as GC content and extraction and library preparation biases result in non-uniform sampling of genomes.
This should therefore be reflected in the simulated datasets.
You may also want to add in some reads from unknown microbes or from human contamination, to add to the authenticity of the simulated samples. </p><p class="css-p9c5cf"><figure class="gatsby-resp-image-figure css-cm98ls">
    <span class="gatsby-resp-image-wrapper" style="position:relative;display:block;margin-left:auto;margin-right:auto;max-width:1140px">
      <span class="gatsby-resp-image-background-image" style="padding-bottom:28.421052631578945%;position:relative;bottom:0;left:0;display:block"></span>
  <img class="gatsby-resp-image-image css-1nh7vc8" alt="Metagenomics simulation" title="How to simulate datasets for metagenomics abundance analysis." src="/static/f8fc62a5e97d8f09b16fad5ae4badffe/b5cea/metagenomics.png" srcSet="/static/f8fc62a5e97d8f09b16fad5ae4badffe/0e2fe/metagenomics.png 285w,/static/f8fc62a5e97d8f09b16fad5ae4badffe/432e7/metagenomics.png 570w,/static/f8fc62a5e97d8f09b16fad5ae4badffe/b5cea/metagenomics.png 1140w,/static/f8fc62a5e97d8f09b16fad5ae4badffe/09ede/metagenomics.png 1710w,/static/f8fc62a5e97d8f09b16fad5ae4badffe/d50e7/metagenomics.png 2280w,/static/f8fc62a5e97d8f09b16fad5ae4badffe/836a3/metagenomics.png 2905w" sizes="(max-width: 1140px) 100vw, 1140px" style="width:100%;height:100%;margin:0;vertical-align:middle;position:absolute;top:0;left:0" loading="lazy" decoding="async"/>
    </span>
    <figcaption class="gatsby-resp-image-figcaption css-13jeo2h">How to simulate datasets for metagenomics abundance analysis.</figcaption>
  </figure></p><p class="css-p9c5cf">If the above approach is not feasible due to lack of datasets, then another option is to throw in a bit of the first simulation approach! </p><h3 id="i-need-to-test-how-my-pipeline-performs-with-lower-allele-frequency" class="css-11m2vzb"><a href="#i-need-to-test-how-my-pipeline-performs-with-lower-allele-frequency" class="css-3mxrie">I need to test how my pipeline performs with lower allele frequency<svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 512 512" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg"><path d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"></path></svg></a></h3><p class="css-p9c5cf">Use the second approach to mix reads from two different samples at various proportions, resulting in variants with altered allele frequencies.</p><p class="css-p9c5cf"><figure class="gatsby-resp-image-figure css-cm98ls">
    <span class="gatsby-resp-image-wrapper" style="position:relative;display:block;margin-left:auto;margin-right:auto;max-width:1140px">
      <span class="gatsby-resp-image-background-image" style="padding-bottom:46.666666666666664%;position:relative;bottom:0;left:0;display:block"></span>
  <img class="gatsby-resp-image-image css-1nh7vc8" alt="Low allele frequency simulation" title="How to simulate datasets to test performance with low allele frequency." src="/static/c917a4bf3c23f93e453ea08c45df5521/b5cea/lowAF.png" srcSet="/static/c917a4bf3c23f93e453ea08c45df5521/0e2fe/lowAF.png 285w,/static/c917a4bf3c23f93e453ea08c45df5521/432e7/lowAF.png 570w,/static/c917a4bf3c23f93e453ea08c45df5521/b5cea/lowAF.png 1140w,/static/c917a4bf3c23f93e453ea08c45df5521/09ede/lowAF.png 1710w,/static/c917a4bf3c23f93e453ea08c45df5521/d50e7/lowAF.png 2280w,/static/c917a4bf3c23f93e453ea08c45df5521/4fe3b/lowAF.png 2901w" sizes="(max-width: 1140px) 100vw, 1140px" style="width:100%;height:100%;margin:0;vertical-align:middle;position:absolute;top:0;left:0" loading="lazy" decoding="async"/>
    </span>
    <figcaption class="gatsby-resp-image-figcaption css-13jeo2h">How to simulate datasets to test performance with low allele frequency.</figcaption>
  </figure></p><h3 id="i-need-to-test-my-short-variant-caller-on-a-specific-set-of-novel-variants" class="css-11m2vzb"><a href="#i-need-to-test-my-short-variant-caller-on-a-specific-set-of-novel-variants" class="css-3mxrie">I need to test my short variant caller on a specific set of novel variants<svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 512 512" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg"><path d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"></path></svg></a></h3><p class="css-p9c5cf">This requires simulating new variants through either of the approaches.
As we mentioned before, these each have their own drawbacks so it’s worth considering which suits your needs best before deciding on one.</p><p class="css-p9c5cf">When using the first approach you can simulate both the reference sequence and novel variants in proportions corresponding to any desired allele frequencies for the study.</p><p class="css-p9c5cf"><figure class="gatsby-resp-image-figure css-cm98ls">
    <span class="gatsby-resp-image-wrapper" style="position:relative;display:block;margin-left:auto;margin-right:auto;max-width:1140px">
      <span class="gatsby-resp-image-background-image" style="padding-bottom:30.87719298245614%;position:relative;bottom:0;left:0;display:block"></span>
  <img class="gatsby-resp-image-image css-1nh7vc8" alt="Novel variant simulation" title="How to simulate datasets to test variant calling on novel variants." src="/static/2521265c47a41482a6fcbc5ad49071ec/b5cea/novel.png" srcSet="/static/2521265c47a41482a6fcbc5ad49071ec/0e2fe/novel.png 285w,/static/2521265c47a41482a6fcbc5ad49071ec/432e7/novel.png 570w,/static/2521265c47a41482a6fcbc5ad49071ec/b5cea/novel.png 1140w,/static/2521265c47a41482a6fcbc5ad49071ec/09ede/novel.png 1710w,/static/2521265c47a41482a6fcbc5ad49071ec/d50e7/novel.png 2280w,/static/2521265c47a41482a6fcbc5ad49071ec/0a61a/novel.png 3822w" sizes="(max-width: 1140px) 100vw, 1140px" style="width:100%;height:100%;margin:0;vertical-align:middle;position:absolute;top:0;left:0" loading="lazy" decoding="async"/>
    </span>
    <figcaption class="gatsby-resp-image-figcaption css-13jeo2h">How to simulate datasets to test variant calling on novel variants.</figcaption>
  </figure></p><p class="css-p9c5cf">For the second approach, you’ll need to acquire real sequencing dataset for which all true variant positions are reliably known (such that these true variants don’t conflate the study).
You can then swap bases at the required positions in a portion of reads covering a simulated variant position. </p><h3 id="i-need-to-test-the-performance-of-my-breakpoint-based-copy-number-variant-caller" class="css-11m2vzb"><a href="#i-need-to-test-the-performance-of-my-breakpoint-based-copy-number-variant-caller" class="css-3mxrie">I need to test the performance of my breakpoint-based copy number variant caller<svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 512 512" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg"><path d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"></path></svg></a></h3><p class="css-p9c5cf">Breakpoints and complex structural rearrangements cannot be modelled easily with the second approach.
It is possible in theory to do so by cutting and joining reads together, but this quickly becomes a headache in practice!
That leaves us with the first approach probably being our best option.
Copy number variants or translocations can be added into a genome sequence and processed to create reads that represent those structural rearrangements.</p><h3 id="i-need-to-test-my-tumour-evolution-analysis-pipeline" class="css-11m2vzb"><a href="#i-need-to-test-my-tumour-evolution-analysis-pipeline" class="css-3mxrie">I need to test my tumour evolution analysis pipeline<svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 512 512" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg"><path d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"></path></svg></a></h3><p class="css-p9c5cf">We’ve come to our last scenario!
Tumours are constantly evolving, with sub-populations of cells forming that contain distinct sets of genetic mutations.
Tumour evolution analysis covers a range of techniques that can be applied to tumour sequencing datasets, and which aim to characterise the distinct cell populations and uncover the evolutionary relationships between them. </p><p class="css-p9c5cf">But how to simulate suitable benchmarking datasets for them?</p><p class="css-p9c5cf">This requires generating datasets that represent a phylogenetic tree of evolving cell populations, all descended from the tumour’s cell of origin.
Simulating such a dataset can be achieved by using either of the approaches, and extending many of the above scenarios in such a way that once one cell population is simulated, subsequent populations are then simulated by acquiring all variants from their parent cell population and incorporating additional new variants. </p><p class="css-p9c5cf">You’ll probably want to first simulate a germline “cell population”, followed by a cell of origin “cell population”.
After that, divergence of cell populations can be achieved by adding distinct sets of variants to two populations originating from the same parent, and shorter or larger evolutionary distances can be reflected by altering the number of new variants added. </p><p class="css-p9c5cf">Once the reads for each cell population are created, they can be merged at required frequencies.
Different samples can be created that contain varying proportions of each cell population to reflect different tumour sampling regions or time points through therapy.</p><p class="css-p9c5cf"><figure class="gatsby-resp-image-figure css-cm98ls">
    <span class="gatsby-resp-image-wrapper" style="position:relative;display:block;margin-left:auto;margin-right:auto;max-width:1140px">
      <span class="gatsby-resp-image-background-image" style="padding-bottom:44.56140350877193%;position:relative;bottom:0;left:0;display:block"></span>
  <img class="gatsby-resp-image-image css-1nh7vc8" alt="Evolution simulation" title="How to simulate datasets to test tumour evolution analysis." src="/static/ba8a7f04711c4f5c288428c2e6077fe8/b5cea/evolution.png" srcSet="/static/ba8a7f04711c4f5c288428c2e6077fe8/0e2fe/evolution.png 285w,/static/ba8a7f04711c4f5c288428c2e6077fe8/432e7/evolution.png 570w,/static/ba8a7f04711c4f5c288428c2e6077fe8/b5cea/evolution.png 1140w,/static/ba8a7f04711c4f5c288428c2e6077fe8/09ede/evolution.png 1710w,/static/ba8a7f04711c4f5c288428c2e6077fe8/d50e7/evolution.png 2280w,/static/ba8a7f04711c4f5c288428c2e6077fe8/4fe3b/evolution.png 2901w" sizes="(max-width: 1140px) 100vw, 1140px" style="width:100%;height:100%;margin:0;vertical-align:middle;position:absolute;top:0;left:0" loading="lazy" decoding="async"/>
    </span>
    <figcaption class="gatsby-resp-image-figcaption css-13jeo2h">How to simulate datasets to test tumour evolution analysis.</figcaption>
  </figure></p><p class="css-p9c5cf">It is not necessary to merge, or even generate reads, from every stage of the phylogenetic tree.
In tumours, cell populations can die off from being outcompeted by others or as a result of treatments, or they might only be present in certain regions of the tumour.</p><style data-emotion-css="cgfyok">.css-cgfyok{padding:0;margin:0;margin-bottom:1rem;margin-left:0;margin-right:0;font-style:italic;border-left-style:solid;border-left-color:var(--theme-ui-colors-alphaLight,#90C5E7);border-left-width:3px;padding-left:1rem;padding-right:1rem;padding:0;margin:0;margin-bottom:1rem;margin-left:0;margin-right:0;font-style:italic;border-left-style:solid;border-left-color:var(--theme-ui-colors-alphaLight,#90C5E7);border-left-width:3px;padding-left:1rem;padding-right:1rem;}@media screen and (min-width:640px){.css-cgfyok{margin-left:2rem;margin-right:2rem;padding-left:2rem;padding-right:2rem;}}.css-cgfyok p{color:var(--theme-ui-colors-text,#cbd5e0);}.css-cgfyok.translation{font-size:1rem;}@media screen and (min-width:640px){.css-cgfyok{margin-left:2rem;margin-right:2rem;padding-left:2rem;padding-right:2rem;}}.css-cgfyok p{color:var(--theme-ui-colors-text,#cbd5e0);}.css-cgfyok.translation{font-size:1rem;}</style><blockquote class="css-cgfyok"><p class="css-p9c5cf">A similar approach to simulating tumours might be useful for simulating microbial evolution.</p></blockquote><p class="css-p9c5cf">One final point to mention about this scenario: an important feature of tumour evolution is that the genome can become very messy and complex.
For example, copy number variants can overlap other copy number variants, and short variants can occur on one or multiple copies of a multiplied region depending on whether the short variant occurred in time before or after the copy number variant.
These features can make it even more challenging to analyse tumour evolution.
So if the analysis you’re wanting to benchmark could be affected by this complexity then it’s important to include them in your simulated datasets.</p><h2 id="simulation-tool-examples" class="css-1ophxj7"><a href="#simulation-tool-examples" class="css-3n7dj1">Simulation tool examples<svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 512 512" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg"><path d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"></path></svg></a></h2><p class="css-p9c5cf">You’re hopefully now aware of how to simulate sequencing datasets in various different scenarios.
Many more applications will exist and by understanding the considerations needed for the above, you will have an idea of how to decide on the best approach for others.</p><ul class="css-16v5hhi"><li class="css-1o70w8g">Icarust - a complete MinKNOW simulator including squiggle and basecall simulation, <a href="https://academic.oup.com/bioinformatics/advance-article/doi/10.1093/bioinformatics/btae141/7628125" class="css-14yhzo">Munro et al. 2024</a></li><li class="css-1o70w8g">NanoSIM - Simulates nanopore long-read sequencing data, <a href="https://doi.org/10.1093/gigascience/gix010" class="css-14yhzo">Yang et al. 2017</a></li><li class="css-1o70w8g">DeepSimulator1.5 - Simulates nanopore long-read sequencing data, via simulating the raw electrical signals <a href="https://doi.org/10.1093/bioinformatics/btz963" class="css-14yhzo">Wang et al. 2020</a></li><li class="css-1o70w8g">A recent review of other short-read simulators - <a href="https://doi.org/10.1038/s41437-022-00577-3" class="css-14yhzo">Milhaven and Pfeifer 2023</a></li><li class="css-1o70w8g">CAMISIM - Metagenomics simulator, incorporating various other short and long-read simulators, <a href="https://doi.org/10.1186/s40168-019-0633-6" class="css-14yhzo">Fritz 2019 et al.</a></li><li class="css-1o70w8g">HeteroGenesis - Simulates genome sequences for complex somatic samples, <a href="https://doi.org/10.1093/bioinformatics/bty1063" class="css-14yhzo">Tanner et al. 2019</a></li></ul><h2 id="what-next" class="css-1ophxj7"><a href="#what-next" class="css-3n7dj1">What next?<svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 512 512" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg"><path d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"></path></svg></a></h2><p class="css-p9c5cf">For a guide on how to perform benchmarking once you have your simulated datasets, see our recent blog post on <a href="https://epi2me.nanoporetech.com/benchmarking-performance/" class="css-14yhzo">Benchmarking performance with sensitivity, specificity, precision and recall</a>.</p><p class="css-p9c5cf">Happy simulating!</p><div class="css-vurnku"><style data-emotion-css="1n0vkfc">.css-1n0vkfc{box-sizing:border-box;margin:0;min-width:0;color:gray;margin:0;margin-top:0.5rem;margin-bottom:0.5rem;border:0;border-bottom:1px solid;padding:0;margin:0;margin-bottom:2rem;border:0;border-bottom:1px solid;margin-top:2rem;color:var(--theme-ui-colors-omegaLight,#001A21);}</style><hr class="css-1n0vkfc"/><style data-emotion-css="1y5vlh1">.css-1y5vlh1{box-sizing:border-box;margin:0;min-width:0;-webkit-align-items:flex-start;-webkit-box-align:flex-start;-ms-flex-align:flex-start;align-items:flex-start;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}@media screen and (min-width:640px){.css-1y5vlh1{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;}}.css-1y5vlh1 > div + div{margin-top:1rem;-webkit-flex-basis:50%;-ms-flex-preferred-size:50%;flex-basis:50%;-webkit-box-pack:end;-webkit-justify-content:flex-end;-ms-flex-pack:end;justify-content:flex-end;}@media screen and (min-width:640px){.css-1y5vlh1 > div + div{margin-top:0;}}</style><div class="css-1y5vlh1"><style data-emotion-css="zkfaav">.css-zkfaav{box-sizing:border-box;margin:0;min-width:0;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}</style><div class="css-zkfaav"><style data-emotion-css="sq2rma">.css-sq2rma{box-sizing:border-box;margin:0;min-width:0;font-family:'Inter',sans-serif;font-weight:bold;line-height:1.33;display:block;color:var(--theme-ui-colors-heading,#edf2f7);font-weight:600;-webkit-text-decoration:none;text-decoration:none;margin-bottom:1rem;font-size:0.875rem;color:var(--theme-ui-colors-omegaDark,#e2e8f0);margin-right:1rem;margin-bottom:0;}</style><h2 class="css-sq2rma">Tags</h2><style data-emotion-css="1sine8z">.css-1sine8z{box-sizing:border-box;margin:0;min-width:0;}.css-1sine8z a{margin-bottom:0.25rem;margin-right:0.25rem;}</style><div class="css-1sine8z"><style data-emotion-css="1v5cak8">.css-1v5cak8{box-sizing:border-box;margin:0;min-width:0;display:inline-block;vertical-align:baseline;font-size:0.75rem;font-weight:700;white-space:nowrap;padding-left:0.25rem;padding-right:0.25rem;border-radius:2px;color:var(--theme-ui-colors-white,#fff);background-color:primary;display:inline-block;border-radius:0.5rem;-webkit-text-decoration:none;text-decoration:none;font-weight:600;vertical-align:middle;padding-left:1rem;padding-right:1rem;padding-top:0.25rem;padding-bottom:0.25rem;-webkit-transition:all 250ms ease;transition:all 250ms ease;background-color:var(--theme-ui-colors-omegaLight,#001A21);color:var(--theme-ui-colors-omegaDark,#e2e8f0);}.css-1v5cak8:hover{color:var(--theme-ui-colors-omegaLight,#001A21);background-color:var(--theme-ui-colors-omegaDark,#e2e8f0);}</style><a class="css-1v5cak8" href="/tag/benchmarking/">#benchmarking</a><a class="css-1v5cak8" href="/tag/performance/">#performance</a></div></div><!--$--><div class="css-zkfaav"><style data-emotion-css="aov8rj">.css-aov8rj{box-sizing:border-box;margin:0;min-width:0;font-family:'Inter',sans-serif;font-weight:bold;line-height:1.33;display:block;color:var(--theme-ui-colors-heading,#edf2f7);font-weight:600;-webkit-text-decoration:none;text-decoration:none;margin-bottom:1rem;font-size:0.875rem;color:var(--theme-ui-colors-omegaDark,#e2e8f0);margin-right:0.25rem;margin-bottom:0;}</style><h2 class="css-aov8rj">Share</h2><style data-emotion-css="1bboj5n">.css-1bboj5n{box-sizing:border-box;margin:0;min-width:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;padding:0.25rem;width:32px;height:32px;color:inherit;background-color:transparent;border:none;border-radius:4px;display:inline-block;min-width:auto;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:space-evenly;-webkit-justify-content:space-evenly;-ms-flex-pack:space-evenly;justify-content:space-evenly;outline:0;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-tap-highlight-color:transparent;-webkit-touch-callout:none;-webkit-user-select:none;-khtml-user-select:none;-moz-user-select:none;-ms-user-select:none;border-radius:9999px;cursor:pointer;border-width:3px;border-style:solid;-webkit-transition:all 250ms ease;transition:all 250ms ease;box-sizing:initial;color:var(--theme-ui-colors-omega,#a0aec0);border-color:transparent;vertical-align:middle;width:24px;height:24px;padding:0.5rem;color:#4267B2;}.css-1bboj5n +a,.css-1bboj5n + button{margin-left:0.5rem;}.css-1bboj5n svg{width:24px;height:24px;}@media (hover:hover) and (pointer:fine){.css-1bboj5n:hover{background-color:var(--theme-ui-colors-alpha,#0084A9);color:var(--theme-ui-colors-white,#fff);}.css-1bboj5n:hover svg{fill:var(--theme-ui-colors-white,#fff);}}.css-1bboj5n:before{display:none;}</style><span class="css-1bboj5n"><button aria-label="facebook" class="react-share__ShareButton" style="background-color:transparent;border:none;padding:0;font:inherit;color:inherit;cursor:pointer"><svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 512 512" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg"><path d="M512 256C512 114.6 397.4 0 256 0S0 114.6 0 256C0 376 82.7 476.8 194.2 504.5V334.2H141.4V256h52.8V222.3c0-87.1 39.4-127.5 125-127.5c16.2 0 44.2 3.2 55.7 6.4V172c-6-.6-16.5-1-29.6-1c-42 0-58.2 15.9-58.2 57.2V256h83.6l-14.4 78.2H287V510.1C413.8 494.8 512 386.9 512 256h0z"></path></svg></button></span><style data-emotion-css="1cecmy5">.css-1cecmy5{box-sizing:border-box;margin:0;min-width:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;padding:0.25rem;width:32px;height:32px;color:inherit;background-color:transparent;border:none;border-radius:4px;display:inline-block;min-width:auto;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:space-evenly;-webkit-justify-content:space-evenly;-ms-flex-pack:space-evenly;justify-content:space-evenly;outline:0;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-tap-highlight-color:transparent;-webkit-touch-callout:none;-webkit-user-select:none;-khtml-user-select:none;-moz-user-select:none;-ms-user-select:none;border-radius:9999px;cursor:pointer;border-width:3px;border-style:solid;-webkit-transition:all 250ms ease;transition:all 250ms ease;box-sizing:initial;color:var(--theme-ui-colors-omega,#a0aec0);border-color:transparent;vertical-align:middle;width:24px;height:24px;padding:0.5rem;color:#000000;}.css-1cecmy5 +a,.css-1cecmy5 + button{margin-left:0.5rem;}.css-1cecmy5 svg{width:24px;height:24px;}@media (hover:hover) and (pointer:fine){.css-1cecmy5:hover{background-color:var(--theme-ui-colors-alpha,#0084A9);color:var(--theme-ui-colors-white,#fff);}.css-1cecmy5:hover svg{fill:var(--theme-ui-colors-white,#fff);}}.css-1cecmy5:before{display:none;}</style><span class="css-1cecmy5"><button aria-label="twitter" class="react-share__ShareButton" style="background-color:transparent;border:none;padding:0;font:inherit;color:inherit;cursor:pointer"><svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 448 512" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg"><path d="M64 32C28.7 32 0 60.7 0 96V416c0 35.3 28.7 64 64 64H384c35.3 0 64-28.7 64-64V96c0-35.3-28.7-64-64-64H64zm297.1 84L257.3 234.6 379.4 396H283.8L209 298.1 123.3 396H75.8l111-126.9L69.7 116h98l67.7 89.5L313.6 116h47.5zM323.3 367.6L153.4 142.9H125.1L296.9 367.6h26.3z"></path></svg></button></span><style data-emotion-css="znygzb">.css-znygzb{box-sizing:border-box;margin:0;min-width:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;padding:0.25rem;width:32px;height:32px;color:inherit;background-color:transparent;border:none;border-radius:4px;display:inline-block;min-width:auto;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:space-evenly;-webkit-justify-content:space-evenly;-ms-flex-pack:space-evenly;justify-content:space-evenly;outline:0;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-tap-highlight-color:transparent;-webkit-touch-callout:none;-webkit-user-select:none;-khtml-user-select:none;-moz-user-select:none;-ms-user-select:none;border-radius:9999px;cursor:pointer;border-width:3px;border-style:solid;-webkit-transition:all 250ms ease;transition:all 250ms ease;box-sizing:initial;color:var(--theme-ui-colors-omega,#a0aec0);border-color:transparent;vertical-align:middle;width:24px;height:24px;padding:0.5rem;color:#2867B2;}.css-znygzb +a,.css-znygzb + button{margin-left:0.5rem;}.css-znygzb svg{width:24px;height:24px;}@media (hover:hover) and (pointer:fine){.css-znygzb:hover{background-color:var(--theme-ui-colors-alpha,#0084A9);color:var(--theme-ui-colors-white,#fff);}.css-znygzb:hover svg{fill:var(--theme-ui-colors-white,#fff);}}.css-znygzb:before{display:none;}</style><span class="css-znygzb"><button quote="Simulating sequencing datasets" aria-label="linkedin" class="react-share__ShareButton" style="background-color:transparent;border:none;padding:0;font:inherit;color:inherit;cursor:pointer"><svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 448 512" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg"><path d="M100.28 448H7.4V148.9h92.88zM53.79 108.1C24.09 108.1 0 83.5 0 53.8a53.79 53.79 0 0 1 107.58 0c0 29.7-24.1 54.3-53.79 54.3zM447.9 448h-92.68V302.4c0-34.7-.7-79.2-48.29-79.2-48.29 0-55.69 37.7-55.69 76.7V448h-92.78V148.9h89.08v40.8h1.3c12.4-23.5 42.69-48.3 87.88-48.3 94 0 111.28 61.9 111.28 142.3V448z"></path></svg></button></span><style data-emotion-css="ucr5gj">.css-ucr5gj{box-sizing:border-box;margin:0;min-width:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;padding:0.25rem;width:32px;height:32px;color:inherit;background-color:transparent;border:none;border-radius:4px;display:inline-block;min-width:auto;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:space-evenly;-webkit-justify-content:space-evenly;-ms-flex-pack:space-evenly;justify-content:space-evenly;outline:0;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-tap-highlight-color:transparent;-webkit-touch-callout:none;-webkit-user-select:none;-khtml-user-select:none;-moz-user-select:none;-ms-user-select:none;border-radius:9999px;cursor:pointer;border-width:3px;border-style:solid;-webkit-transition:all 250ms ease;transition:all 250ms ease;box-sizing:initial;color:var(--theme-ui-colors-omega,#a0aec0);border-color:transparent;vertical-align:middle;width:24px;height:24px;padding:0.5rem;color:#718096;}.css-ucr5gj +a,.css-ucr5gj + button{margin-left:0.5rem;}.css-ucr5gj svg{width:24px;height:24px;}@media (hover:hover) and (pointer:fine){.css-ucr5gj:hover{background-color:var(--theme-ui-colors-alpha,#0084A9);color:var(--theme-ui-colors-white,#fff);}.css-ucr5gj:hover svg{fill:var(--theme-ui-colors-white,#fff);}}.css-ucr5gj:before{display:none;}</style><span class="css-ucr5gj"><button aria-label="email" class="react-share__ShareButton" style="background-color:transparent;border:none;padding:0;font:inherit;color:inherit;cursor:pointer"><svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 512 512" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg"><path d="M48 64C21.5 64 0 85.5 0 112c0 15.1 7.1 29.3 19.2 38.4L236.8 313.6c11.4 8.5 27 8.5 38.4 0L492.8 150.4c12.1-9.1 19.2-23.3 19.2-38.4c0-26.5-21.5-48-48-48L48 64zM0 176L0 384c0 35.3 28.7 64 64 64l384 0c35.3 0 64-28.7 64-64l0-208L294.4 339.2c-22.8 17.1-54 17.1-76.8 0L0 176z"></path></svg></button></span></div><!--/$--></div></div></div><style data-emotion-css="1b24wpc">.css-1b24wpc{box-sizing:border-box;margin:0;min-width:0;min-width:auto;border-top-style:solid;border-top-color:transparent;border-top-width:2px;height:0;margin-top:1rem;margin-bottom:1rem;}@media screen and (min-width:640px){.css-1b24wpc{margin-top:2rem;margin-bottom:2rem;}}</style><div class="css-1b24wpc"></div></div><style data-emotion-css="1dgwrt9">.css-1dgwrt9{box-sizing:border-box;margin:0;min-width:0;width:100%;max-width:1400px;margin-left:auto;margin-right:auto;display:none;-webkit-flex-basis:33.333333%;-ms-flex-preferred-size:33.333333%;flex-basis:33.333333%;padding-left:4rem;margin-right:0;}@media screen and (min-width:900px){.css-1dgwrt9{display:block;}}</style><div class="css-1dgwrt9"><div class="css-1teua8m"><div class="css-15qiiq2"><style data-emotion-css="1lx7oj">.css-1lx7oj{box-sizing:border-box;margin:0;min-width:0;text-align:center;}</style><div class="css-1lx7oj"><style data-emotion-css="9ixkre">.css-9ixkre{box-sizing:border-box;margin:0;min-width:0;color:var(--theme-ui-colors-alpha,#0084A9);-webkit-text-decoration:none;text-decoration:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;}.css-9ixkre:visited{color:var(--theme-ui-colors-alpha,#0084A9);}.css-9ixkre:hover{color:var(--theme-ui-colors-alphaDark,#a3bffa);}</style><a class="css-9ixkre" href="/author/georgette-tanner/"><style data-emotion-css="10n2utx">.css-10n2utx{box-sizing:border-box;margin:0;min-width:0;font-family:'Inter',sans-serif;font-weight:bold;line-height:1.33;display:block;color:var(--theme-ui-colors-heading,#edf2f7);font-weight:bold;-webkit-text-decoration:none;text-decoration:none;margin-bottom:1rem;font-size:1.125rem;}</style><h2 class="css-10n2utx">Georgette Tanner</h2></a><style data-emotion-css="ky1utg">.css-ky1utg{box-sizing:border-box;margin:0;min-width:0;font-family:'Inter',sans-serif;font-weight:bold;line-height:1.33;display:block;color:var(--theme-ui-colors-heading,#edf2f7);font-weight:600;-webkit-text-decoration:none;text-decoration:none;margin-bottom:1rem;font-size:1rem;color:var(--theme-ui-colors-omegaDark,#e2e8f0);}</style><h2 class="css-ky1utg">Clinical Bioinformatics Software Developer</h2></div></div></div><div class="css-1b24wpc"></div><style data-emotion-css="1dvm86v">.css-1dvm86v{box-sizing:border-box;margin:0;min-width:0;width:100%;max-width:1400px;margin-left:auto;margin-right:auto;position:-webkit-sticky;position:sticky;top:2rem;}</style><div class="css-1dvm86v"><div class="css-1teua8m"><style data-emotion-css="jw76fk">.css-jw76fk{box-sizing:border-box;margin:0;min-width:0;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-align-items:end;-webkit-box-align:end;-ms-flex-align:end;align-items:end;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}</style><div class="css-jw76fk"><style data-emotion-css="gw3otf">.css-gw3otf{box-sizing:border-box;margin:0;min-width:0;font-family:'Inter',sans-serif;font-weight:bold;line-height:1.33;display:block;color:var(--theme-ui-colors-heading,#edf2f7);font-weight:bold;-webkit-text-decoration:none;text-decoration:none;margin-bottom:1rem;font-size:1.5rem;color:var(--theme-ui-colors-omegaDark,#e2e8f0);margin-left:auto;margin-right:auto;margin-bottom:0;}</style><h2 class="css-gw3otf">Table Of Contents</h2></div><style data-emotion-css="16umaqc">.css-16umaqc{box-sizing:border-box;margin:0;min-width:0;min-width:auto;border-top-style:solid;border-top-color:transparent;border-top-width:2px;height:0;margin-top:0.25rem;margin-bottom:0.25rem;}@media screen and (min-width:640px){.css-16umaqc{margin-top:0.5rem;margin-bottom:0.5rem;}}</style><div class="css-16umaqc"></div><div class="css-15qiiq2"><style data-emotion-css="3qwsrh">.css-3qwsrh{box-sizing:border-box;margin:0;min-width:0;color:var(--theme-ui-colors-omegaDark,#e2e8f0);-webkit-text-decoration:none;text-decoration:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;font-weight:normal;margin-bottom:0.5rem;padding:0;display:block;}.css-3qwsrh:visited{color:var(--theme-ui-colors-omegaDark,#e2e8f0);}.css-3qwsrh:hover{color:var(--theme-ui-colors-alpha,#0084A9);}.css-3qwsrh svg{vertical-align:middle;display:inline-block;margin-right:0.5rem;margin-bottom:0;}.css-3qwsrh + .css-3qwsrh{margin-top:1rem;}</style><a class="css-3qwsrh" href="/simulating-datasets/#general-approaches"><style data-emotion-css="d6l97b">.css-d6l97b{box-sizing:border-box;margin:0;min-width:0;-webkit-align-items:baseline;-webkit-box-align:baseline;-ms-flex-align:baseline;align-items:baseline;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}</style><div class="css-d6l97b"><style data-emotion-css="1it3nzu">.css-1it3nzu{box-sizing:border-box;margin:0;min-width:0;font-family:'Inter',sans-serif;font-weight:bold;line-height:1.33;display:block;color:var(--theme-ui-colors-heading,#edf2f7);font-weight:600;-webkit-text-decoration:none;text-decoration:none;margin-bottom:1rem;font-size:0.875rem;color:var(--theme-ui-colors-alpha,#0084A9);display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;background-color:var(--theme-ui-colors-omegaLighter,#1a202c);border-radius:9999px;width:25px;height:25px;margin-right:0.5rem;margin-bottom:0;}</style><div class="css-1it3nzu">1</div><style data-emotion-css="1td11jg">.css-1td11jg{box-sizing:border-box;margin:0;min-width:0;-webkit-flex:1;-ms-flex:1;flex:1;line-height:1.33;margin-bottom:0;}</style><div class="css-1td11jg">General approaches</div></div></a><a class="css-3qwsrh" href="/simulating-datasets/#which-approach-do-i-use"><div class="css-d6l97b"><div class="css-1it3nzu">2</div><div class="css-1td11jg">Which approach do I use?</div></div></a><a class="css-3qwsrh" href="/simulating-datasets/#simulation-tool-examples"><div class="css-d6l97b"><div class="css-1it3nzu">3</div><div class="css-1td11jg">Simulation tool examples</div></div></a><a class="css-3qwsrh" href="/simulating-datasets/#what-next"><div class="css-d6l97b"><div class="css-1it3nzu">4</div><div class="css-1td11jg">What next?</div></div></a></div></div><div class="css-1b24wpc"></div><div class="css-1teua8m"><div class="css-jw76fk"><h2 class="css-gw3otf">Related Posts</h2></div><div class="css-16umaqc"></div><style data-emotion-css="116hf8s">.css-116hf8s{box-sizing:border-box;margin:0;min-width:0;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-wrap:wrap;-ms-flex-wrap:wrap;flex-wrap:wrap;margin:-0.5rem;}.css-116hf8s .blog_card{max-width:none;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;padding:0.5rem;}</style><div class="css-116hf8s"><style data-emotion-css="1gxfzf0">.css-1gxfzf0{box-sizing:border-box;margin:0;min-width:0;-webkit-flex-basis:100%;-ms-flex-preferred-size:100%;flex-basis:100%;}</style><div class="blog_card css-1gxfzf0"><style data-emotion-css="ui4wn8">.css-ui4wn8{box-sizing:border-box;margin:0;min-width:0;background-color:var(--theme-ui-colors-contentBg,#2d3748);border-radius:1rem;-webkit-transition: -webkit-transform 250ms ease, box-shadow 250ms ease, color 250ms ease;-webkit-transition: transform 250ms ease, box-shadow 250ms ease, color 250ms ease;transition: transform 250ms ease, box-shadow 250ms ease, color 250ms ease;box-shadow:1px 1px 5px 0 rgba(1,1,1,.05);overflow:hidden;height:100%;}@media (hover:hover) and (pointer:fine){.css-ui4wn8:hover{-webkit-transform:translateY(-0.25rem);-ms-transform:translateY(-0.25rem);transform:translateY(-0.25rem);box-shadow: 0px 2px 4px rgba(46,41,51,0.08), 0px 5px 10px rgba(71,63,79,0.16);}}</style><div class="css-ui4wn8"><style data-emotion-css="1opukye">.css-1opukye{box-sizing:border-box;margin:0;min-width:0;-webkit-align-items:stretch;-webkit-box-align:stretch;-ms-flex-align:stretch;align-items:stretch;height:100%;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;border-left-style:solid;border-left-width:5px;border-left-color:var(--theme-ui-colors-alphaLight,#90C5E7);display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}</style><article class="css-1opukye"><style data-emotion-css="1lcwvjb">.css-1lcwvjb{box-sizing:border-box;margin:0;min-width:0;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;-webkit-flex:1;-ms-flex:1;flex:1;min-height:100px;padding:1rem;}</style><div class="css-1lcwvjb"><style data-emotion-css="drfe50">.css-drfe50{box-sizing:border-box;margin:0;min-width:0;font-family:'Inter',sans-serif;font-weight:bold;line-height:1.33;display:block;color:var(--theme-ui-colors-heading,#edf2f7);font-weight:bold;-webkit-text-decoration:none;text-decoration:none;margin-bottom:1rem;display:block;color:var(--theme-ui-colors-heading,#edf2f7);font-weight:600;-webkit-text-decoration:none;text-decoration:none;margin-bottom:0.5rem;font-size:1rem;}</style><a class="css-drfe50" href="/benchmarking-performance/">Benchmarking performance with sensitivity, specificity, precision and recall</a><style data-emotion-css="zcxndt">.css-zcxndt{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}</style><style data-emotion-css="xv6t55">.css-xv6t55{box-sizing:border-box;margin:0;min-width:0;font-size:0.75rem;margin:0;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}</style><div class="css-xv6t55"><style data-emotion-css="1u093if">.css-1u093if{box-sizing:border-box;margin:0;min-width:0;-webkit-flex:1;-ms-flex:1;flex:1;-webkit-flex-wrap:wrap;-ms-flex-wrap:wrap;flex-wrap:wrap;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;color:var(--theme-ui-colors-omega,#a0aec0);display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}</style><div class="css-1u093if"><style data-emotion-css="1urriwz">.css-1urriwz{box-sizing:border-box;margin:0;min-width:0;padding-right:0.5rem;display:inline-block;-webkit-flex-basis:auto;-ms-flex-preferred-size:auto;flex-basis:auto;}</style><div class="css-1urriwz"><a class="css-1jns145" href="/author/chris-alder/"><strong>Chris Alder</strong></a></div><style data-emotion-css="ah2hnh">.css-ah2hnh{box-sizing:border-box;margin:0;min-width:0;overflow:hidden;text-overflow:ellipsis;white-space:nowrap;}.css-ah2hnh > *:not(:last-child) + *:before{content:" ・ ";}</style><div class="css-ah2hnh"><div class="css-12hibeq">February 19, 2024</div><div class="css-12hibeq"><svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 512 512" class="css-uk6cul" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg"><path d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200zm61.8-104.4l-84.9-61.7c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h32c6.6 0 12 5.4 12 12v141.7l66.8 48.6c5.4 3.9 6.5 11.4 2.6 16.8L334.6 349c-3.9 5.3-11.4 6.5-16.8 2.6z"></path></svg> <!-- -->10<!-- --> min</div></div></div></div></div></article></div></div><div class="blog_card css-1gxfzf0"><div class="css-ui4wn8"><article class="css-1opukye"><div class="css-1lcwvjb"><a class="css-drfe50" href="/post-meta-analysis/">Unexpected results, so now what?</a><div class="css-xv6t55"><div class="css-1u093if"><div class="css-1urriwz"><a class="css-1jns145" href="/author/natalia-garcia/"><strong>Natalia Garcia</strong></a></div><div class="css-ah2hnh"><div class="css-12hibeq">July 02, 2024</div><div class="css-12hibeq"><svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 512 512" class="css-uk6cul" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg"><path d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200zm61.8-104.4l-84.9-61.7c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h32c6.6 0 12 5.4 12 12v141.7l66.8 48.6c5.4 3.9 6.5 11.4 2.6 16.8L334.6 349c-3.9 5.3-11.4 6.5-16.8 2.6z"></path></svg> <!-- -->3<!-- --> min</div></div></div></div></div></article></div></div><div class="blog_card css-1gxfzf0"><div class="css-ui4wn8"><article class="css-1opukye"><div class="css-1lcwvjb"><a class="css-drfe50" href="/singularity/">Singularity for bioinformatics</a><div class="css-xv6t55"><div class="css-1u093if"><div class="css-1urriwz"><a class="css-1jns145" href="/author/neil-horner/"><strong>Neil Horner</strong></a></div><div class="css-ah2hnh"><div class="css-12hibeq">February 26, 2024</div><div class="css-12hibeq"><svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 512 512" class="css-uk6cul" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg"><path d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200zm61.8-104.4l-84.9-61.7c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h32c6.6 0 12 5.4 12 12v141.7l66.8 48.6c5.4 3.9 6.5 11.4 2.6 16.8L334.6 349c-3.9 5.3-11.4 6.5-16.8 2.6z"></path></svg> <!-- -->7<!-- --> min</div></div></div></div></div></article></div></div><div class="blog_card css-1gxfzf0"><div class="css-ui4wn8"><article class="css-1opukye"><div class="css-1lcwvjb"><a class="css-drfe50" href="/dot-plots/">Comparing sequences with dot plots</a><div class="css-xv6t55"><div class="css-1u093if"><div class="css-1urriwz"><a class="css-1jns145" href="/author/sarah-griffiths/"><strong>Sarah Griffiths</strong></a></div><div class="css-ah2hnh"><div class="css-12hibeq">February 05, 2024</div><div class="css-12hibeq"><svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 512 512" class="css-uk6cul" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg"><path d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200zm61.8-104.4l-84.9-61.7c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h32c6.6 0 12 5.4 12 12v141.7l66.8 48.6c5.4 3.9 6.5 11.4 2.6 16.8L334.6 349c-3.9 5.3-11.4 6.5-16.8 2.6z"></path></svg> <!-- -->6<!-- --> min</div></div></div></div></div></article></div></div><div class="blog_card css-1gxfzf0"><div class="css-ui4wn8"><article class="css-1opukye"><div class="css-1lcwvjb"><a class="css-drfe50" href="/how-to-meta-offline/">How to build and use databases to run wf-metagenomics and wf-16s offline</a><div class="css-xv6t55"><div class="css-1u093if"><div class="css-1urriwz"><a class="css-1jns145" href="/author/natalia-garcia/"><strong>Natalia Garcia</strong></a></div><div class="css-ah2hnh"><div class="css-12hibeq">October 17, 2023</div><div class="css-12hibeq"><svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 512 512" class="css-uk6cul" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg"><path d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200zm61.8-104.4l-84.9-61.7c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h32c6.6 0 12 5.4 12 12v141.7l66.8 48.6c5.4 3.9 6.5 11.4 2.6 16.8L334.6 349c-3.9 5.3-11.4 6.5-16.8 2.6z"></path></svg> <!-- -->7<!-- --> min</div></div></div></div></div></article></div></div><div class="blog_card css-1gxfzf0"><div class="css-ui4wn8"><article class="css-1opukye"><div class="css-1lcwvjb"><a class="css-drfe50" href="/how-to-exits/">How to interpret exit codes</a><div class="css-xv6t55"><div class="css-1u093if"><div class="css-1urriwz"><a class="css-1jns145" href="/author/sam-nicholls/"><strong>Sam Nicholls</strong></a></div><div class="css-ah2hnh"><div class="css-12hibeq">October 06, 2023</div><div class="css-12hibeq"><svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 512 512" class="css-uk6cul" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg"><path d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200zm61.8-104.4l-84.9-61.7c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h32c6.6 0 12 5.4 12 12v141.7l66.8 48.6c5.4 3.9 6.5 11.4 2.6 16.8L334.6 349c-3.9 5.3-11.4 6.5-16.8 2.6z"></path></svg> <!-- -->4<!-- --> min</div></div></div></div></div></article></div></div></div></div></div></div></div></div></div></div><style data-emotion-css="1ry1o7">.css-1ry1o7{box-sizing:border-box;margin:0;min-width:0;position:relative;background-color:var(--theme-ui-colors-footerBg,#2d3748);margin-top:4rem;}@media screen and (min-width:640px){.css-1ry1o7{margin-top:8rem;}}</style><div class="css-1ry1o7"><style data-emotion-css="3fl8q0">.css-3fl8q0{box-sizing:border-box;margin:0;min-width:0;width:100%;max-width:1400px;margin-left:auto;margin-right:auto;padding-left:1rem;padding-right:1rem;}@media screen and (min-width:640px){.css-3fl8q0{padding-left:2rem;padding-right:2rem;}}</style><div class="css-3fl8q0"><style data-emotion-css="q7bxia">.css-q7bxia{box-sizing:border-box;margin:0;min-width:0;-webkit-flex-direction:column-reverse;-ms-flex-direction:column-reverse;flex-direction:column-reverse;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding-top:4rem;padding-bottom:4rem;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}@media screen and (min-width:640px){.css-q7bxia{-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-align-items:flex-start;-webkit-box-align:flex-start;-ms-flex-align:flex-start;align-items:flex-start;}}</style><div class="css-q7bxia"><div class="css-vurnku"><style data-emotion-css="12brje6">.css-12brje6{box-sizing:border-box;margin:0;min-width:0;font-family:'Inter',sans-serif;font-weight:bold;line-height:1.33;display:block;color:var(--theme-ui-colors-heading,#edf2f7);font-weight:600;-webkit-text-decoration:none;text-decoration:none;margin-bottom:1rem;font-size:1rem;display:none;}@media screen and (min-width:640px){.css-12brje6{display:block;}}</style><p class="css-12brje6">Quick Links</p><style data-emotion-css="j4toqg">.css-j4toqg{box-sizing:border-box;margin:0;min-width:0;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;-webkit-flex-wrap:wrap;-ms-flex-wrap:wrap;flex-wrap:wrap;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}@media screen and (min-width:640px){.css-j4toqg{-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;-webkit-flex-wrap:nowrap;-ms-flex-wrap:nowrap;flex-wrap:nowrap;}}</style><div class="css-j4toqg"><style data-emotion-css="1xoppkd">.css-1xoppkd{box-sizing:border-box;margin:0;min-width:0;color:inherit;-webkit-text-decoration:none;text-decoration:none;font-weight:700;display:inline-block;color:var(--theme-ui-colors-omegaDark,#e2e8f0);-webkit-text-decoration:none;text-decoration:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;padding:0.25rem;margin:0;margin-left:0.5rem;margin-right:0.5rem;}.css-1xoppkd:hover,.css-1xoppkd:focus,.css-1xoppkd.active{color:primary;}.css-1xoppkd:visited{color:var(--theme-ui-colors-omegaDark,#e2e8f0);}.css-1xoppkd:hover{color:var(--theme-ui-colors-alpha,#0084A9);}.css-1xoppkd svg{display:block;margin-left:auto;margin-right:auto;margin-bottom:0.25rem;}@media screen and (min-width:640px){.css-1xoppkd{color:var(--theme-ui-colors-omegaDark,#e2e8f0);-webkit-text-decoration:none;text-decoration:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;font-weight:normal;margin-bottom:0.5rem;padding:0;}.css-1xoppkd:visited{color:var(--theme-ui-colors-omegaDark,#e2e8f0);}.css-1xoppkd:hover{color:var(--theme-ui-colors-alpha,#0084A9);}.css-1xoppkd svg{vertical-align:middle;display:inline-block;margin-right:0.5rem;margin-bottom:0;}}</style><a class="css-1xoppkd" href="/wfindex/">Workflows</a><a class="css-1xoppkd" href="/dataindex/">Open Data</a><a class="css-1xoppkd" href="/contactus/">Contact</a></div></div><div class="css-vurnku"><p class="css-12brje6">Related Links</p><div class="css-j4toqg"><a href="https://www.nanoporetech.com" class="css-1xoppkd">Oxford Nanopore Technologies</a><a href="https://community.nanoporetech.com" class="css-1xoppkd">Nanopore Community</a></div></div><div class="css-vurnku"><p class="css-12brje6">Information</p><div class="css-j4toqg"><a href="https://nanoporetech.com/privacy-policy" class="css-1xoppkd">Privacy Policy</a><a href="https://nanoporetech.com/commercial_information" class="css-1xoppkd">Terms Of Use</a><a href="https://nanoporetech.com/accessibility-statement" class="css-1xoppkd">Accessibility</a><a href="https://epi2me.nanoporetech.com/epi2me-docs/help/information-governance" class="css-1xoppkd">Information Governance</a></div></div><div class="css-vurnku"><p class="css-12brje6">Social Media</p><style data-emotion-css="1ewyfna">.css-1ewyfna{box-sizing:border-box;margin:0;min-width:0;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;-webkit-flex-wrap:wrap;-ms-flex-wrap:wrap;flex-wrap:wrap;margin-bottom:1rem;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}@media screen and (min-width:640px){.css-1ewyfna{-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;-webkit-flex-wrap:nowrap;-ms-flex-wrap:nowrap;flex-wrap:nowrap;margin-bottom:0;}}</style><div class="css-1ewyfna"><a href="https://github.com/epi2me-labs" target="_blank" rel="noopener noreferrer" class="css-1xoppkd"><svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 496 512" color="darkslategray" style="color:darkslategray" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg"><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"></path></svg>github</a><a href="https://twitter.com/nanopore" target="_blank" rel="noopener noreferrer" class="css-1xoppkd"><svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 448 512" color="#000000" style="color:#000000" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg"><path d="M64 32C28.7 32 0 60.7 0 96V416c0 35.3 28.7 64 64 64H384c35.3 0 64-28.7 64-64V96c0-35.3-28.7-64-64-64H64zm297.1 84L257.3 234.6 379.4 396H283.8L209 298.1 123.3 396H75.8l111-126.9L69.7 116h98l67.7 89.5L313.6 116h47.5zM323.3 367.6L153.4 142.9H125.1L296.9 367.6h26.3z"></path></svg>twitter</a></div></div></div><style data-emotion-css="1kq60m0">.css-1kq60m0{box-sizing:border-box;margin:0;min-width:0;text-align:justify;font-size:0.75rem;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}</style><div class="css-1kq60m0">© 2020 - <!-- -->2025<!-- --> Oxford Nanopore Technologies plc. All rights reserved. Registered Office: Gosling Building, Edmund Halley Road, Oxford Science Park, OX4 4DQ, UK | Registered No. 05386273 | VAT No 336942382. Oxford Nanopore Technologies, the Wheel icon, EPI2ME, Flongle, GridION, Metrichor, MinION, MinIT, MinKNOW, Plongle, PromethION, SmidgION, Ubik and VolTRAX are registered trademarks of Oxford Nanopore Technologies plc in various countries. Oxford Nanopore Technologies products are not intended for use for health assessment or to diagnose, treat, mitigate, cure, or prevent any disease or condition.</div></div></div></div></div><div id="gatsby-announcer" style="position:absolute;top:0;width:1px;height:1px;padding:0;overflow:hidden;clip:rect(0, 0, 0, 0);white-space:nowrap;border:0" aria-live="assertive" aria-atomic="true"></div></div><script id="gatsby-script-loader">/*<![CDATA[*/window.pagePath="/simulating-datasets/";/*]]>*/</script><!-- slice-start id="_gatsby-scripts-1" -->
          <script
            id="gatsby-chunk-mapping"
          >
            window.___chunkMapping="{\"app\":[\"/app-e5299dd95f29a2b1e9e8.js\"],\"component---packages-blog-gatsby-blog-core-src-templates-collection-author-js\":[\"/component---packages-blog-gatsby-blog-core-src-templates-collection-author-js-3448fb9c9d4c1118517d.js\"],\"component---packages-blog-gatsby-blog-core-src-templates-collection-category-js\":[\"/component---packages-blog-gatsby-blog-core-src-templates-collection-category-js-abdaea6693b24ea39619.js\"],\"component---packages-blog-gatsby-blog-core-src-templates-collection-tag-js\":[\"/component---packages-blog-gatsby-blog-core-src-templates-collection-tag-js-8c488abaf6981a0300b9.js\"],\"component---packages-blog-gatsby-blog-core-src-templates-post-js\":[\"/component---packages-blog-gatsby-blog-core-src-templates-post-js-85a172238f5632cc349b.js\"],\"component---packages-blog-gatsby-blog-core-src-templates-posts-js\":[\"/component---packages-blog-gatsby-blog-core-src-templates-posts-js-5d662045fec866e24cc0.js\"],\"component---packages-blog-gatsby-blog-pages-src-pages-404-jsx\":[\"/component---packages-blog-gatsby-blog-pages-src-pages-404-jsx-17d3a8129168db68f3f7.js\"],\"component---packages-blog-gatsby-blog-pages-src-pages-contact-jsx\":[\"/component---packages-blog-gatsby-blog-pages-src-pages-contact-jsx-03873ed27310ba0a03d9.js\"],\"component---src-elegantstack-gatsby-blog-pages-pages-authors-jsx\":[\"/component---src-elegantstack-gatsby-blog-pages-pages-authors-jsx-701faaa3a02b66aff24c.js\"]}";
          </script>
        <script>window.___webpackCompilationHash="76d052b6bc341774fb35";</script><script src="/webpack-runtime-94434a90a03ae83de523.js" async></script><script src="/framework-0036e75a995d4c43b433.js" async></script><script src="/42765b2c-5818486b9a078264305c.js" async></script><script src="/d5395874-c018db10fddae203fe2f.js" async></script><script src="/0b7ad3e3-a5fac5c9a2111bf7c01b.js" async></script><script src="/0e71760c-df498ea720998f01e457.js" async></script><script src="/9bab7026-13cd376ab764f85375d0.js" async></script><script src="/7d643cfb-7869ac6f4c26d56f3fcb.js" async></script><script src="/20f2b20d-f56c50cf9ba66911c99e.js" async></script><script src="/a453d9fc-bc79ff0fc9101f4418bd.js" async></script><script src="/181af4e5-100316cb4051599e84fe.js" async></script><script src="/8d698823-0b1a6dfa0a965bf27529.js" async></script><script src="/aa120c45-f77d0b9a01f6e436ffc7.js" async></script><script src="/09793ad7-697c61984027b11129d8.js" async></script><script src="/a139dce8-00526713a2cb87f548e0.js" async></script><script src="/e5aaf470-fb88eb50724e25ef8763.js" async></script><script src="/app-e5299dd95f29a2b1e9e8.js" async></script><!-- slice-end id="_gatsby-scripts-1" --></body></html>